2023-08-30 07:40:15,928 [DEBUG]  Available web server routes: 
/conversations/<conversation_id:path>/messages     POST                           add_message
/conversations/<conversation_id:path>/tracker/events POST                           append_events
/webhooks/rasa                                     GET                            custom_webhook_RasaChatInput.health
/webhooks/rasa/webhook                             POST                           custom_webhook_RasaChatInput.receive
/webhooks/rest                                     GET                            custom_webhook_RestInput.health
/webhooks/rest/webhook                             POST                           custom_webhook_RestInput.receive
/model/test/intents                                POST                           evaluate_intents
/model/test/stories                                POST                           evaluate_stories
/conversations/<conversation_id:path>/execute      POST                           execute_action
/domain                                            GET                            get_domain
/                                                  GET                            hello
/model                                             PUT                            load_model
/model/parse                                       POST                           parse
/conversations/<conversation_id:path>/predict      POST                           predict
/conversations/<conversation_id:path>/tracker/events PUT                            replace_events
/conversations/<conversation_id:path>/story        GET                            retrieve_story
/conversations/<conversation_id:path>/tracker      GET                            retrieve_tracker
/status                                            GET                            status
/model/predict                                     POST                           tracker_predict
/model/train                                       POST                           train
/conversations/<conversation_id:path>/trigger_intent POST                           trigger_intent
/model                                             DELETE                         unload_model
/version                                           GET                            version
2023-08-30 07:40:15,946 [INFO ]  Starting Rasa server on http://0.0.0.0:5005
2023-08-30 07:40:15,947 [DEBUG]  Using the default number of Sanic workers (1).
2023-08-30 07:40:16,750 [DEBUG]  Starting new HTTPS connection (1): api.segment.io:443
2023-08-30 07:40:17,592 [DEBUG]  https://api.segment.io:443 "POST /v1/track HTTP/1.1" 200 21
2023-08-30 07:40:17,635 [DEBUG]  Connected to InMemoryTrackerStore.
2023-08-30 07:40:17,636 [DEBUG]  Connected to lock store 'InMemoryLockStore'.
2023-08-30 07:40:17,637 [DEBUG]  Instantiated NLG to 'TemplatedNaturalLanguageGenerator'.
2023-08-30 07:40:17,638 [INFO ]  Loading model /app/models/20230830-073139-augmenting-convolution.tar.gz...
2023-08-30 07:40:18,258 [DEBUG]  Extracted model to '/tmp/tmp4jdhwols'.
2023-08-30 07:40:18,261 [DEBUG]  Node 'nlu_message_converter' loading 'NLUMessageConverter.load' and kwargs: '{}'.
2023-08-30 07:40:18,261 [DEBUG]  Node 'run_WhitespaceTokenizer0' loading 'WhitespaceTokenizer.load' and kwargs: '{}'.
2023-08-30 07:40:18,264 [DEBUG]  Node 'run_RegexFeaturizer1' loading 'RegexFeaturizer.load' and kwargs: '{}'.
2023-08-30 07:40:18,264 [DEBUG]  Resource 'train_RegexFeaturizer1' was requested for reading.
2023-08-30 07:40:18,265 [DEBUG]  Node 'run_RegexEntityExtractor2' loading 'RegexEntityExtractor.load' and kwargs: '{}'.
2023-08-30 07:40:18,265 [DEBUG]  Resource 'train_RegexEntityExtractor2' was requested for reading.
2023-08-30 07:40:18,266 [DEBUG]  Node 'run_LexicalSyntacticFeaturizer3' loading 'LexicalSyntacticFeaturizer.load' and kwargs: '{}'.
2023-08-30 07:40:18,267 [DEBUG]  Resource 'train_LexicalSyntacticFeaturizer3' was requested for reading.
2023-08-30 07:40:18,269 [DEBUG]  Node 'run_CountVectorsFeaturizer4' loading 'CountVectorsFeaturizer.load' and kwargs: '{}'.
2023-08-30 07:40:18,270 [DEBUG]  Resource 'train_CountVectorsFeaturizer4' was requested for reading.
2023-08-30 07:40:18,273 [DEBUG]  Node 'run_CountVectorsFeaturizer5' loading 'CountVectorsFeaturizer.load' and kwargs: '{}'.
2023-08-30 07:40:18,273 [DEBUG]  Resource 'train_CountVectorsFeaturizer5' was requested for reading.
2023-08-30 07:40:18,283 [DEBUG]  Node 'run_DIETClassifier6' loading 'DIETClassifier.load' and kwargs: '{}'.
2023-08-30 07:40:18,284 [DEBUG]  Resource 'train_DIETClassifier6' was requested for reading.
2023-08-30 07:40:18,285 [DEBUG]  Loading the model from /tmp/tmpgpf27c5x/train_DIETClassifier6/DIETClassifier.tf_model with finetune_mode=False...
2023-08-30 07:40:18,335 [DEBUG]  Following metrics will be logged during training: 
2023-08-30 07:40:18,336 [DEBUG]    t_loss (total loss)
2023-08-30 07:40:18,337 [DEBUG]    i_acc (intent acc)
2023-08-30 07:40:18,338 [DEBUG]    i_loss (intent loss)
2023-08-30 07:40:18,339 [DEBUG]    e_f1 (entity f1)
2023-08-30 07:40:18,340 [DEBUG]    e_loss (entity loss)
2023-08-30 07:40:46,947 [DEBUG]  Finished loading the model.
2023-08-30 07:40:46,948 [DEBUG]  Node 'run_EntitySynonymMapper7' loading 'EntitySynonymMapper.load' and kwargs: '{}'.
2023-08-30 07:40:46,948 [DEBUG]  Resource 'train_EntitySynonymMapper7' was requested for reading.
2023-08-30 07:40:46,949 [DEBUG]  Node 'run_ResponseSelector8' loading 'ResponseSelector.load' and kwargs: '{}'.
2023-08-30 07:40:46,950 [DEBUG]  Resource 'train_ResponseSelector8' was requested for reading.
2023-08-30 07:40:46,951 [DEBUG]  Failed to load ABCMeta from model storage. Resource 'train_ResponseSelector8' doesn't exist.
2023-08-30 07:40:46,951 [DEBUG]  Resource 'train_ResponseSelector8' was requested for reading.
2023-08-30 07:40:46,952 [DEBUG]  Failed to load ResponseSelector from model storage. Resource 'train_ResponseSelector8' doesn't exist.
2023-08-30 07:40:46,952 [DEBUG]  Node 'run_FallbackClassifier9' loading 'FallbackClassifier.load' and kwargs: '{}'.
2023-08-30 07:40:46,953 [DEBUG]  Node 'run_DucklingEntityExtractor10' loading 'DucklingEntityExtractor.load' and kwargs: '{}'.
2023-08-30 07:40:46,953 [DEBUG]  Node 'run_RegexMessageHandler' loading 'RegexMessageHandler.load' and kwargs: '{}'.
2023-08-30 07:40:46,953 [DEBUG]  Node 'domain_provider' loading 'DomainProvider.load' and kwargs: '{}'.
2023-08-30 07:40:46,954 [DEBUG]  Resource 'domain_provider' was requested for reading.
2023-08-30 07:40:46,971 [DEBUG]  Node 'run_MemoizationPolicy0' loading 'MemoizationPolicy.load' and kwargs: '{}'.
2023-08-30 07:40:46,971 [DEBUG]  Resource 'train_MemoizationPolicy0' was requested for reading.
2023-08-30 07:40:46,972 [DEBUG]  Node 'run_RulePolicy1' loading 'RulePolicy.load' and kwargs: '{}'.
2023-08-30 07:40:46,973 [DEBUG]  Resource 'train_RulePolicy1' was requested for reading.
2023-08-30 07:40:46,974 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' loading 'UnexpecTEDIntentPolicy.load' and kwargs: '{}'.
2023-08-30 07:40:46,974 [DEBUG]  Resource 'train_UnexpecTEDIntentPolicy2' was requested for reading.
2023-08-30 07:40:46,977 [DEBUG]  Loading the model from /tmp/tmpgpf27c5x/train_UnexpecTEDIntentPolicy2/unexpected_intent_policy.tf_model with finetune_mode=False...
2023-08-30 07:40:57,385 [DEBUG]  Finished loading the model.
2023-08-30 07:40:57,385 [WARNI]  The UnexpecTED Intent Policy is currently experimental and might change or be removed in the future ðŸ”¬ Please share your feedback on it in the forum (https://forum.rasa.com) to help us make this feature ready for production.
2023-08-30 07:40:57,386 [DEBUG]  Node 'run_TEDPolicy3' loading 'TEDPolicy.load' and kwargs: '{}'.
2023-08-30 07:40:57,386 [DEBUG]  Resource 'train_TEDPolicy3' was requested for reading.
2023-08-30 07:40:57,389 [DEBUG]  Loading the model from /tmp/tmpgpf27c5x/train_TEDPolicy3/ted_policy.tf_model with finetune_mode=False...
2023-08-30 07:41:08,303 [DEBUG]  Finished loading the model.
2023-08-30 07:41:08,304 [DEBUG]  Node 'rule_only_data_provider' loading 'RuleOnlyDataProvider.load' and kwargs: '{}'.
2023-08-30 07:41:08,304 [DEBUG]  Resource 'train_RulePolicy1' was requested for reading.
2023-08-30 07:41:08,305 [DEBUG]  Node 'select_prediction' loading 'DefaultPolicyPredictionEnsemble.load' and kwargs: '{}'.
2023-08-30 07:41:08,312 [INFO ]  Rasa server is up and running.
2023-08-30 07:41:08,315 [INFO ]  Enabling coroutine debugging. Loop id 120230896.
2023-08-30 07:42:22,346 [DEBUG]  Issuing ticket for conversation 'user'.
2023-08-30 07:42:22,346 [DEBUG]  Acquiring lock for conversation 'user'.
2023-08-30 07:42:22,348 [DEBUG]  Acquired lock for conversation 'user'.
2023-08-30 07:42:22,348 [DEBUG]  Could not find tracker for conversation ID 'user'.
2023-08-30 07:42:22,349 [DEBUG]  Starting a new session for conversation ID 'user'.
2023-08-30 07:42:22,351 [DEBUG]  Policy prediction ended with events '[]'.
2023-08-30 07:42:22,351 [DEBUG]  Action 'action_session_start' ended with events '[<rasa.shared.core.events.SessionStarted object at 0x7f7c5529ebb0>, ActionExecuted(action: action_listen, policy: None, confidence: None)]'.
2023-08-30 07:42:22,352 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 07:42:22,352 [DEBUG]  Running graph with inputs: {'__message__': [<rasa.core.channels.channel.UserMessage object at 0x7f7c7efcaaf0>]}, targets: ['run_RegexMessageHandler'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 07:42:22,354 [DEBUG]  Node 'nlu_message_converter' running 'NLUMessageConverter.convert_user_message'.
2023-08-30 07:42:22,354 [DEBUG]  Node 'run_WhitespaceTokenizer0' running 'WhitespaceTokenizer.process'.
2023-08-30 07:42:22,356 [DEBUG]  Node 'run_RegexFeaturizer1' running 'RegexFeaturizer.process'.
2023-08-30 07:42:22,359 [DEBUG]  Node 'run_RegexEntityExtractor2' running 'RegexEntityExtractor.process'.
2023-08-30 07:42:22,361 [DEBUG]  Node 'run_LexicalSyntacticFeaturizer3' running 'LexicalSyntacticFeaturizer.process'.
2023-08-30 07:42:22,362 [DEBUG]  Node 'run_CountVectorsFeaturizer4' running 'CountVectorsFeaturizer.process'.
2023-08-30 07:42:22,363 [DEBUG]  Node 'run_CountVectorsFeaturizer5' running 'CountVectorsFeaturizer.process'.
2023-08-30 07:42:22,364 [DEBUG]  Node 'run_DIETClassifier6' running 'DIETClassifier.process'.
2023-08-30 07:42:22,678 [DEBUG]  Node 'run_EntitySynonymMapper7' running 'EntitySynonymMapper.process'.
2023-08-30 07:42:22,678 [DEBUG]  Node 'run_ResponseSelector8' running 'ResponseSelector.process'.
2023-08-30 07:42:22,678 [DEBUG]  There is no trained model for 'ResponseSelector': The component is either not trained or didn't receive enough training data.
2023-08-30 07:42:22,679 [DEBUG]  Adding following selector key to message property: default
2023-08-30 07:42:22,679 [DEBUG]  Node 'run_FallbackClassifier9' running 'FallbackClassifier.process'.
2023-08-30 07:42:22,680 [DEBUG]  Node 'run_DucklingEntityExtractor10' running 'DucklingEntityExtractor.process'.
2023-08-30 07:42:22,682 [DEBUG]  Starting new HTTP connection (1): localhost:8000
2023-08-30 07:42:22,683 [ERROR]  Failed to connect to duckling http server. Make sure the duckling server is running/healthy/not stale and the proper host and port are set in the configuration. More information on how to run the server can be found on github: https://github.com/facebook/duckling#quickstart Error: HTTPConnectionPool(host='localhost', port=8000): Max retries exceeded with url: /parse (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f7c5527b250>: Failed to establish a new connection: [Errno 111] Connection refused'))
2023-08-30 07:42:22,684 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 07:42:22,684 [DEBUG]  Node 'run_RegexMessageHandler' running 'RegexMessageHandler.process'.
2023-08-30 07:42:22,685 [DEBUG]  Received user message 'Who is Nguyen Ba Tinh ?' with intent '{'name': 'ask_about_historicalFigure', 'confidence': 1.0}' and entities '[{'entity': 'class', 'start': 0, 'end': 3, 'confidence_entity': 0.9999799728393555, 'value': 'culturaltourism:HistoricalFigure', 'extractor': 'DIETClassifier', 'processors': ['EntitySynonymMapper']}, {'entity': 'object', 'start': 7, 'end': 21, 'confidence_entity': 0.9999908208847046, 'value': 'culturaltourism:Nguyá»…n_BÃ¡_TÄ©nh', 'extractor': 'DIETClassifier', 'processors': ['EntitySynonymMapper']}]'
2023-08-30 07:42:22,686 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 07:42:22,687 [DEBUG]  Logged UserUtterance - tracker now has 4 events.
2023-08-30 07:42:22,688 [DEBUG]  Validating extracted slots: 
2023-08-30 07:42:22,689 [DEBUG]  Default action 'action_extract_slots' was executed, resulting in 0 events: 
2023-08-30 07:42:22,689 [DEBUG]  Running graph with inputs: {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x7f7c55464910>}, targets: ['select_prediction'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 07:42:22,690 [DEBUG]  Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
2023-08-30 07:42:22,691 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 07:42:22,691 [DEBUG]  Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
2023-08-30 07:42:22,692 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
2023-08-30 07:42:22,692 [DEBUG]  There is no memorised next action
2023-08-30 07:42:22,693 [DEBUG]  Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
2023-08-30 07:42:22,693 [DEBUG]  Current tracker state:
[state 1] user text: Who is Nguyen Ba Tinh ? | previous action name: action_listen
2023-08-30 07:42:22,694 [DEBUG]  There is no applicable rule.
2023-08-30 07:42:22,695 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
2023-08-30 07:42:22,695 [DEBUG]  There is a rule for the next action 'action_one_condition'.
2023-08-30 07:42:22,695 [DEBUG]  Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
2023-08-30 07:42:22,701 [DEBUG]  TED predicted 'action_listing' based on user intent.
2023-08-30 07:42:22,702 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
2023-08-30 07:42:23,348 [DEBUG]  Querying for intent `ask_about_historicalFigure`.
2023-08-30 07:42:23,349 [DEBUG]  Query intent index 2 not found in label thresholds - {5: 1.5548791}. Check for `action_unlikely_intent` prediction will be skipped.
2023-08-30 07:42:23,351 [DEBUG]  Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
2023-08-30 07:42:23,352 [DEBUG]  Made prediction using user intent.
2023-08-30 07:42:23,353 [DEBUG]  Added `DefinePrevUserUtteredFeaturization(False)` event.
2023-08-30 07:42:23,353 [DEBUG]  Predicted next action using RulePolicy.
2023-08-30 07:42:23,354 [DEBUG]  Predicted next action 'action_one_condition' with confidence 1.00.
2023-08-30 07:42:23,355 [DEBUG]  Calling action endpoint to run action 'action_one_condition'.
2023-08-30 07:42:23,361 [ERROR]  Failed to run custom action 'action_one_condition'. Couldn't connect to the server at 'http://localhost:5055/webhook'. Is the server running? Error: Cannot connect to host localhost:5055 ssl:default [Connection refused]
2023-08-30 07:42:23,361 [ERROR]  Encountered an exception while running action 'action_one_condition'.Bot will continue, but the actions events are lost. Please check the logs of your action server for more information.
Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 980, in _wrap_create_connection
    return await self._loop.create_connection(*args, **kwargs)  # type: ignore[return-value]  # noqa
  File "uvloop/loop.pyx", line 2039, in create_connection
  File "uvloop/loop.pyx", line 2016, in uvloop.loop.Loop.create_connection
ConnectionRefusedError: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/actions/action.py", line 741, in run
    response: Any = await self.action_endpoint.request(
  File "/opt/venv/lib/python3.8/site-packages/rasa/utils/endpoints.py", line 164, in request
    async with session.request(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/client.py", line 1141, in __aenter__
    self._resp = await self._coro
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/client.py", line 536, in _request
    conn = await self._connector.connect(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 540, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 901, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 1206, in _create_direct_connection
    raise last_exc
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 1175, in _create_direct_connection
    transp, proto = await self._wrap_create_connection(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 988, in _wrap_create_connection
    raise client_error(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host localhost:5055 ssl:default [Connection refused]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/processor.py", line 874, in _run_action
    events = await action.run(
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/actions/action.py", line 776, in run
    raise RasaException(
rasa.shared.exceptions.RasaException: Failed to execute custom action 'action_one_condition'. Couldn't connect to the server at 'http://localhost:5055/webhook.
2023-08-30 07:42:23,362 [DEBUG]  Policy prediction ended with events '[<rasa.shared.core.events.DefinePrevUserUtteredFeaturization object at 0x7f7c5527beb0>]'.
2023-08-30 07:42:23,363 [DEBUG]  Action 'action_one_condition' ended with events '[]'.
2023-08-30 07:42:23,363 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 07:42:23,366 [DEBUG]  Running graph with inputs: {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x7f7c55464910>}, targets: ['select_prediction'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 07:42:23,367 [DEBUG]  Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
2023-08-30 07:42:23,367 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 07:42:23,368 [DEBUG]  Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
2023-08-30 07:42:23,368 [DEBUG]  Current tracker state:
[state 1] previous action name: action_listen
2023-08-30 07:42:23,370 [DEBUG]  There is no memorised next action
2023-08-30 07:42:23,372 [DEBUG]  Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
2023-08-30 07:42:23,379 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 2] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
2023-08-30 07:42:23,380 [DEBUG]  There is a rule for the next action 'action_listen'.
2023-08-30 07:42:23,383 [DEBUG]  Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
2023-08-30 07:42:23,388 [DEBUG]  TED predicted 'action_listen' based on user intent.
2023-08-30 07:42:23,389 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
2023-08-30 07:42:23,389 [DEBUG]  Skipping predictions for UnexpecTEDIntentPolicy as either there is no event of type `UserUttered`, event's intent is new and not in domain or there is an event of type `ActionExecuted` after the last `UserUttered`.
2023-08-30 07:42:23,390 [DEBUG]  Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
2023-08-30 07:42:23,391 [DEBUG]  Predicted next action using RulePolicy.
2023-08-30 07:42:23,392 [DEBUG]  Predicted next action 'action_listen' with confidence 1.00.
2023-08-30 07:42:23,392 [DEBUG]  Policy prediction ended with events '[]'.
2023-08-30 07:42:23,392 [DEBUG]  Action 'action_listen' ended with events '[]'.
2023-08-30 07:42:23,393 [DEBUG]  Deleted lock for conversation 'user'.
2023-08-30 07:43:14,260 [DEBUG]  Issuing ticket for conversation 'user'.
2023-08-30 07:43:14,261 [DEBUG]  Acquiring lock for conversation 'user'.
2023-08-30 07:43:14,261 [DEBUG]  Acquired lock for conversation 'user'.
2023-08-30 07:43:14,262 [DEBUG]  Recreating tracker for id 'user'
2023-08-30 07:43:14,263 [DEBUG]  Running graph with inputs: {'__message__': [<rasa.core.channels.channel.UserMessage object at 0x7f7c7efca970>]}, targets: ['run_RegexMessageHandler'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 07:43:14,264 [DEBUG]  Node 'nlu_message_converter' running 'NLUMessageConverter.convert_user_message'.
2023-08-30 07:43:14,265 [DEBUG]  Node 'run_WhitespaceTokenizer0' running 'WhitespaceTokenizer.process'.
2023-08-30 07:43:14,265 [DEBUG]  Node 'run_RegexFeaturizer1' running 'RegexFeaturizer.process'.
2023-08-30 07:43:14,266 [DEBUG]  Node 'run_RegexEntityExtractor2' running 'RegexEntityExtractor.process'.
2023-08-30 07:43:14,267 [DEBUG]  Node 'run_LexicalSyntacticFeaturizer3' running 'LexicalSyntacticFeaturizer.process'.
2023-08-30 07:43:14,268 [DEBUG]  Node 'run_CountVectorsFeaturizer4' running 'CountVectorsFeaturizer.process'.
2023-08-30 07:43:14,269 [DEBUG]  Node 'run_CountVectorsFeaturizer5' running 'CountVectorsFeaturizer.process'.
2023-08-30 07:43:14,270 [DEBUG]  Node 'run_DIETClassifier6' running 'DIETClassifier.process'.
2023-08-30 07:43:14,281 [DEBUG]  Node 'run_EntitySynonymMapper7' running 'EntitySynonymMapper.process'.
2023-08-30 07:43:14,282 [DEBUG]  Node 'run_ResponseSelector8' running 'ResponseSelector.process'.
2023-08-30 07:43:14,282 [DEBUG]  There is no trained model for 'ResponseSelector': The component is either not trained or didn't receive enough training data.
2023-08-30 07:43:14,283 [DEBUG]  Adding following selector key to message property: default
2023-08-30 07:43:14,283 [DEBUG]  Node 'run_FallbackClassifier9' running 'FallbackClassifier.process'.
2023-08-30 07:43:14,284 [DEBUG]  Node 'run_DucklingEntityExtractor10' running 'DucklingEntityExtractor.process'.
2023-08-30 07:43:14,286 [DEBUG]  Starting new HTTP connection (1): localhost:8000
2023-08-30 07:43:14,287 [ERROR]  Failed to connect to duckling http server. Make sure the duckling server is running/healthy/not stale and the proper host and port are set in the configuration. More information on how to run the server can be found on github: https://github.com/facebook/duckling#quickstart Error: HTTPConnectionPool(host='localhost', port=8000): Max retries exceeded with url: /parse (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f7c551b2130>: Failed to establish a new connection: [Errno 111] Connection refused'))
2023-08-30 07:43:14,288 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 07:43:14,289 [DEBUG]  Node 'run_RegexMessageHandler' running 'RegexMessageHandler.process'.
2023-08-30 07:43:14,289 [DEBUG]  Received user message 'Who is Nguyen Ba Tinh ?' with intent '{'name': 'ask_about_historicalFigure', 'confidence': 1.0}' and entities '[{'entity': 'class', 'start': 0, 'end': 3, 'confidence_entity': 0.9999799728393555, 'value': 'culturaltourism:HistoricalFigure', 'extractor': 'DIETClassifier', 'processors': ['EntitySynonymMapper']}, {'entity': 'object', 'start': 7, 'end': 21, 'confidence_entity': 0.9999908208847046, 'value': 'culturaltourism:Nguyá»…n_BÃ¡_TÄ©nh', 'extractor': 'DIETClassifier', 'processors': ['EntitySynonymMapper']}]'
2023-08-30 07:43:14,291 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 07:43:14,292 [DEBUG]  Logged UserUtterance - tracker now has 8 events.
2023-08-30 07:43:14,292 [DEBUG]  Validating extracted slots: 
2023-08-30 07:43:14,293 [DEBUG]  Default action 'action_extract_slots' was executed, resulting in 0 events: 
2023-08-30 07:43:14,294 [DEBUG]  Running graph with inputs: {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x7f7c5529e6a0>}, targets: ['select_prediction'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 07:43:14,295 [DEBUG]  Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
2023-08-30 07:43:14,296 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 07:43:14,297 [DEBUG]  Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
2023-08-30 07:43:14,297 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
2023-08-30 07:43:14,298 [DEBUG]  There is no memorised next action
2023-08-30 07:43:14,299 [DEBUG]  Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
2023-08-30 07:43:14,299 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 2] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 3] user text: Who is Nguyen Ba Tinh ? | previous action name: action_listen
2023-08-30 07:43:14,300 [DEBUG]  There is no applicable rule.
2023-08-30 07:43:14,300 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 2] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 3] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
2023-08-30 07:43:14,301 [DEBUG]  There is a rule for the next action 'action_one_condition'.
2023-08-30 07:43:14,302 [DEBUG]  Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
2023-08-30 07:43:14,307 [DEBUG]  TED predicted 'action_listing' based on user intent.
2023-08-30 07:43:14,308 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
2023-08-30 07:43:14,313 [DEBUG]  Querying for intent `ask_about_historicalFigure`.
2023-08-30 07:43:14,313 [DEBUG]  Query intent index 2 not found in label thresholds - {5: 1.5548791}. Check for `action_unlikely_intent` prediction will be skipped.
2023-08-30 07:43:14,314 [DEBUG]  Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
2023-08-30 07:43:14,315 [DEBUG]  Made prediction using user intent.
2023-08-30 07:43:14,315 [DEBUG]  Added `DefinePrevUserUtteredFeaturization(False)` event.
2023-08-30 07:43:14,315 [DEBUG]  Predicted next action using RulePolicy.
2023-08-30 07:43:14,316 [DEBUG]  Predicted next action 'action_one_condition' with confidence 1.00.
2023-08-30 07:43:14,317 [DEBUG]  Calling action endpoint to run action 'action_one_condition'.
2023-08-30 07:43:14,321 [ERROR]  Failed to run custom action 'action_one_condition'. Couldn't connect to the server at 'http://localhost:5055/webhook'. Is the server running? Error: Cannot connect to host localhost:5055 ssl:default [Connection refused]
2023-08-30 07:43:14,321 [ERROR]  Encountered an exception while running action 'action_one_condition'.Bot will continue, but the actions events are lost. Please check the logs of your action server for more information.
Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 980, in _wrap_create_connection
    return await self._loop.create_connection(*args, **kwargs)  # type: ignore[return-value]  # noqa
  File "uvloop/loop.pyx", line 2039, in create_connection
  File "uvloop/loop.pyx", line 2016, in uvloop.loop.Loop.create_connection
ConnectionRefusedError: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/actions/action.py", line 741, in run
    response: Any = await self.action_endpoint.request(
  File "/opt/venv/lib/python3.8/site-packages/rasa/utils/endpoints.py", line 164, in request
    async with session.request(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/client.py", line 1141, in __aenter__
    self._resp = await self._coro
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/client.py", line 536, in _request
    conn = await self._connector.connect(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 540, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 901, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 1206, in _create_direct_connection
    raise last_exc
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 1175, in _create_direct_connection
    transp, proto = await self._wrap_create_connection(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 988, in _wrap_create_connection
    raise client_error(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host localhost:5055 ssl:default [Connection refused]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/processor.py", line 874, in _run_action
    events = await action.run(
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/actions/action.py", line 776, in run
    raise RasaException(
rasa.shared.exceptions.RasaException: Failed to execute custom action 'action_one_condition'. Couldn't connect to the server at 'http://localhost:5055/webhook.
2023-08-30 07:43:14,322 [DEBUG]  Policy prediction ended with events '[<rasa.shared.core.events.DefinePrevUserUtteredFeaturization object at 0x7f7c551b2730>]'.
2023-08-30 07:43:14,322 [DEBUG]  Action 'action_one_condition' ended with events '[]'.
2023-08-30 07:43:14,323 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 07:43:14,323 [DEBUG]  Running graph with inputs: {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x7f7c5529e6a0>}, targets: ['select_prediction'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 07:43:14,324 [DEBUG]  Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
2023-08-30 07:43:14,325 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 07:43:14,325 [DEBUG]  Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
2023-08-30 07:43:14,326 [DEBUG]  Current tracker state:
[state 1] previous action name: action_listen
2023-08-30 07:43:14,328 [DEBUG]  There is no memorised next action
2023-08-30 07:43:14,329 [DEBUG]  Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
2023-08-30 07:43:14,330 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 2] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 3] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 4] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
2023-08-30 07:43:14,330 [DEBUG]  There is a rule for the next action 'action_listen'.
2023-08-30 07:43:14,332 [DEBUG]  Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
2023-08-30 07:43:14,337 [DEBUG]  TED predicted 'action_listen' based on user intent.
2023-08-30 07:43:14,338 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
2023-08-30 07:43:14,338 [DEBUG]  Skipping predictions for UnexpecTEDIntentPolicy as either there is no event of type `UserUttered`, event's intent is new and not in domain or there is an event of type `ActionExecuted` after the last `UserUttered`.
2023-08-30 07:43:14,339 [DEBUG]  Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
2023-08-30 07:43:14,340 [DEBUG]  Predicted next action using RulePolicy.
2023-08-30 07:43:14,341 [DEBUG]  Predicted next action 'action_listen' with confidence 1.00.
2023-08-30 07:43:14,341 [DEBUG]  Policy prediction ended with events '[]'.
2023-08-30 07:43:14,341 [DEBUG]  Action 'action_listen' ended with events '[]'.
2023-08-30 07:43:14,342 [DEBUG]  Deleted lock for conversation 'user'.
2023-08-30 07:44:25,547 [DEBUG]  Issuing ticket for conversation 'user'.
2023-08-30 07:44:25,548 [DEBUG]  Acquiring lock for conversation 'user'.
2023-08-30 07:44:25,548 [DEBUG]  Acquired lock for conversation 'user'.
2023-08-30 07:44:25,548 [DEBUG]  Recreating tracker for id 'user'
2023-08-30 07:44:25,550 [DEBUG]  Running graph with inputs: {'__message__': [<rasa.core.channels.channel.UserMessage object at 0x7f7c55464d00>]}, targets: ['run_RegexMessageHandler'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 07:44:25,551 [DEBUG]  Node 'nlu_message_converter' running 'NLUMessageConverter.convert_user_message'.
2023-08-30 07:44:25,551 [DEBUG]  Node 'run_WhitespaceTokenizer0' running 'WhitespaceTokenizer.process'.
2023-08-30 07:44:25,552 [DEBUG]  Node 'run_RegexFeaturizer1' running 'RegexFeaturizer.process'.
2023-08-30 07:44:25,553 [DEBUG]  Node 'run_RegexEntityExtractor2' running 'RegexEntityExtractor.process'.
2023-08-30 07:44:25,553 [DEBUG]  Node 'run_LexicalSyntacticFeaturizer3' running 'LexicalSyntacticFeaturizer.process'.
2023-08-30 07:44:25,554 [DEBUG]  Node 'run_CountVectorsFeaturizer4' running 'CountVectorsFeaturizer.process'.
2023-08-30 07:44:25,555 [DEBUG]  Node 'run_CountVectorsFeaturizer5' running 'CountVectorsFeaturizer.process'.
2023-08-30 07:44:25,557 [DEBUG]  Node 'run_DIETClassifier6' running 'DIETClassifier.process'.
2023-08-30 07:44:25,570 [DEBUG]  Node 'run_EntitySynonymMapper7' running 'EntitySynonymMapper.process'.
2023-08-30 07:44:25,571 [DEBUG]  Node 'run_ResponseSelector8' running 'ResponseSelector.process'.
2023-08-30 07:44:25,571 [DEBUG]  There is no trained model for 'ResponseSelector': The component is either not trained or didn't receive enough training data.
2023-08-30 07:44:25,572 [DEBUG]  Adding following selector key to message property: default
2023-08-30 07:44:25,573 [DEBUG]  Node 'run_FallbackClassifier9' running 'FallbackClassifier.process'.
2023-08-30 07:44:25,574 [DEBUG]  Node 'run_DucklingEntityExtractor10' running 'DucklingEntityExtractor.process'.
2023-08-30 07:44:25,577 [DEBUG]  Starting new HTTP connection (1): localhost:8000
2023-08-30 07:44:25,579 [ERROR]  Failed to connect to duckling http server. Make sure the duckling server is running/healthy/not stale and the proper host and port are set in the configuration. More information on how to run the server can be found on github: https://github.com/facebook/duckling#quickstart Error: HTTPConnectionPool(host='localhost', port=8000): Max retries exceeded with url: /parse (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f7c5527b580>: Failed to establish a new connection: [Errno 111] Connection refused'))
2023-08-30 07:44:25,579 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 07:44:25,580 [DEBUG]  Node 'run_RegexMessageHandler' running 'RegexMessageHandler.process'.
2023-08-30 07:44:25,581 [DEBUG]  Received user message 'Tell me about Mac Kinh Vu ?' with intent '{'name': 'ask_about_historicalFigure', 'confidence': 1.0}' and entities '[{'entity': 'object', 'start': 14, 'end': 25, 'confidence_entity': 0.9999877214431763, 'value': 'culturaltourism:Máº¡c_KÃ­nh_VÅ©', 'extractor': 'DIETClassifier', 'processors': ['EntitySynonymMapper']}]'
2023-08-30 07:44:25,581 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 07:44:25,582 [DEBUG]  Logged UserUtterance - tracker now has 12 events.
2023-08-30 07:44:25,582 [DEBUG]  Validating extracted slots: 
2023-08-30 07:44:25,583 [DEBUG]  Default action 'action_extract_slots' was executed, resulting in 0 events: 
2023-08-30 07:44:25,584 [DEBUG]  Running graph with inputs: {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x7f7c551b23d0>}, targets: ['select_prediction'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 07:44:25,585 [DEBUG]  Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
2023-08-30 07:44:25,585 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 07:44:25,586 [DEBUG]  Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
2023-08-30 07:44:25,586 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
2023-08-30 07:44:25,587 [DEBUG]  There is no memorised next action
2023-08-30 07:44:25,588 [DEBUG]  Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
2023-08-30 07:44:25,588 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 2] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 3] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 4] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 5] user text: Tell me about Mac Kinh Vu ? | previous action name: action_listen
2023-08-30 07:44:25,589 [DEBUG]  There is no applicable rule.
2023-08-30 07:44:25,590 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 2] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 3] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 4] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 5] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
2023-08-30 07:44:25,591 [DEBUG]  There is a rule for the next action 'action_one_condition'.
2023-08-30 07:44:25,591 [DEBUG]  Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
2023-08-30 07:44:25,598 [DEBUG]  TED predicted 'action_listen' based on user intent.
2023-08-30 07:44:25,599 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
2023-08-30 07:44:25,603 [DEBUG]  Querying for intent `ask_about_historicalFigure`.
2023-08-30 07:44:25,604 [DEBUG]  Query intent index 2 not found in label thresholds - {5: 1.5548791}. Check for `action_unlikely_intent` prediction will be skipped.
2023-08-30 07:44:25,605 [DEBUG]  Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
2023-08-30 07:44:25,606 [DEBUG]  Made prediction using user intent.
2023-08-30 07:44:25,607 [DEBUG]  Added `DefinePrevUserUtteredFeaturization(False)` event.
2023-08-30 07:44:25,607 [DEBUG]  Predicted next action using RulePolicy.
2023-08-30 07:44:25,608 [DEBUG]  Predicted next action 'action_one_condition' with confidence 1.00.
2023-08-30 07:44:25,609 [DEBUG]  Calling action endpoint to run action 'action_one_condition'.
2023-08-30 07:44:25,613 [ERROR]  Failed to run custom action 'action_one_condition'. Couldn't connect to the server at 'http://localhost:5055/webhook'. Is the server running? Error: Cannot connect to host localhost:5055 ssl:default [Connection refused]
2023-08-30 07:44:25,613 [ERROR]  Encountered an exception while running action 'action_one_condition'.Bot will continue, but the actions events are lost. Please check the logs of your action server for more information.
Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 980, in _wrap_create_connection
    return await self._loop.create_connection(*args, **kwargs)  # type: ignore[return-value]  # noqa
  File "uvloop/loop.pyx", line 2039, in create_connection
  File "uvloop/loop.pyx", line 2016, in uvloop.loop.Loop.create_connection
ConnectionRefusedError: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/actions/action.py", line 741, in run
    response: Any = await self.action_endpoint.request(
  File "/opt/venv/lib/python3.8/site-packages/rasa/utils/endpoints.py", line 164, in request
    async with session.request(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/client.py", line 1141, in __aenter__
    self._resp = await self._coro
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/client.py", line 536, in _request
    conn = await self._connector.connect(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 540, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 901, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 1206, in _create_direct_connection
    raise last_exc
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 1175, in _create_direct_connection
    transp, proto = await self._wrap_create_connection(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 988, in _wrap_create_connection
    raise client_error(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host localhost:5055 ssl:default [Connection refused]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/processor.py", line 874, in _run_action
    events = await action.run(
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/actions/action.py", line 776, in run
    raise RasaException(
rasa.shared.exceptions.RasaException: Failed to execute custom action 'action_one_condition'. Couldn't connect to the server at 'http://localhost:5055/webhook.
2023-08-30 07:44:25,615 [DEBUG]  Policy prediction ended with events '[<rasa.shared.core.events.DefinePrevUserUtteredFeaturization object at 0x7f7c5525a190>]'.
2023-08-30 07:44:25,615 [DEBUG]  Action 'action_one_condition' ended with events '[]'.
2023-08-30 07:44:25,616 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 07:44:25,616 [DEBUG]  Running graph with inputs: {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x7f7c551b23d0>}, targets: ['select_prediction'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 07:44:25,617 [DEBUG]  Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
2023-08-30 07:44:25,618 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 07:44:25,620 [DEBUG]  Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
2023-08-30 07:44:25,621 [DEBUG]  Current tracker state:
[state 1] previous action name: action_listen
2023-08-30 07:44:25,622 [DEBUG]  There is no memorised next action
2023-08-30 07:44:25,624 [DEBUG]  Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
2023-08-30 07:44:25,625 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 2] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 3] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 4] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 5] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
[state 6] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_one_condition
2023-08-30 07:44:25,626 [DEBUG]  There is a rule for the next action 'action_listen'.
2023-08-30 07:44:25,627 [DEBUG]  Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
2023-08-30 07:44:25,632 [DEBUG]  TED predicted 'action_listen' based on user intent.
2023-08-30 07:44:25,632 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
2023-08-30 07:44:25,633 [DEBUG]  Skipping predictions for UnexpecTEDIntentPolicy as either there is no event of type `UserUttered`, event's intent is new and not in domain or there is an event of type `ActionExecuted` after the last `UserUttered`.
2023-08-30 07:44:25,633 [DEBUG]  Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
2023-08-30 07:44:25,634 [DEBUG]  Predicted next action using RulePolicy.
2023-08-30 07:44:25,634 [DEBUG]  Predicted next action 'action_listen' with confidence 1.00.
2023-08-30 07:44:25,635 [DEBUG]  Policy prediction ended with events '[]'.
2023-08-30 07:44:25,635 [DEBUG]  Action 'action_listen' ended with events '[]'.
2023-08-30 07:44:25,636 [DEBUG]  Deleted lock for conversation 'user'.
2023-08-30 07:45:20,564 [DEBUG]  Issuing ticket for conversation 'user'.
2023-08-30 07:45:20,565 [DEBUG]  Acquiring lock for conversation 'user'.
2023-08-30 07:45:20,565 [DEBUG]  Acquired lock for conversation 'user'.
2023-08-30 07:45:20,566 [DEBUG]  Recreating tracker for id 'user'
2023-08-30 07:45:20,568 [DEBUG]  Running graph with inputs: {'__message__': [<rasa.core.channels.channel.UserMessage object at 0x7f7c551b2700>]}, targets: ['run_RegexMessageHandler'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 07:45:20,569 [DEBUG]  Node 'nlu_message_converter' running 'NLUMessageConverter.convert_user_message'.
2023-08-30 07:45:20,571 [DEBUG]  Node 'run_WhitespaceTokenizer0' running 'WhitespaceTokenizer.process'.
2023-08-30 07:45:20,572 [DEBUG]  Node 'run_RegexFeaturizer1' running 'RegexFeaturizer.process'.
2023-08-30 07:45:20,573 [DEBUG]  Node 'run_RegexEntityExtractor2' running 'RegexEntityExtractor.process'.
2023-08-30 07:45:20,574 [DEBUG]  Node 'run_LexicalSyntacticFeaturizer3' running 'LexicalSyntacticFeaturizer.process'.
2023-08-30 07:45:20,575 [DEBUG]  Node 'run_CountVectorsFeaturizer4' running 'CountVectorsFeaturizer.process'.
2023-08-30 07:45:20,577 [DEBUG]  Node 'run_CountVectorsFeaturizer5' running 'CountVectorsFeaturizer.process'.
2023-08-30 07:45:20,579 [DEBUG]  Node 'run_DIETClassifier6' running 'DIETClassifier.process'.
2023-08-30 07:45:20,590 [DEBUG]  Node 'run_EntitySynonymMapper7' running 'EntitySynonymMapper.process'.
2023-08-30 07:45:20,591 [DEBUG]  Node 'run_ResponseSelector8' running 'ResponseSelector.process'.
2023-08-30 07:45:20,591 [DEBUG]  There is no trained model for 'ResponseSelector': The component is either not trained or didn't receive enough training data.
2023-08-30 07:45:20,592 [DEBUG]  Adding following selector key to message property: default
2023-08-30 07:45:20,593 [DEBUG]  Node 'run_FallbackClassifier9' running 'FallbackClassifier.process'.
2023-08-30 07:45:20,595 [DEBUG]  Node 'run_DucklingEntityExtractor10' running 'DucklingEntityExtractor.process'.
2023-08-30 07:45:20,598 [DEBUG]  Starting new HTTP connection (1): localhost:8000
2023-08-30 07:45:20,599 [ERROR]  Failed to connect to duckling http server. Make sure the duckling server is running/healthy/not stale and the proper host and port are set in the configuration. More information on how to run the server can be found on github: https://github.com/facebook/duckling#quickstart Error: HTTPConnectionPool(host='localhost', port=8000): Max retries exceeded with url: /parse (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f7c551a1190>: Failed to establish a new connection: [Errno 111] Connection refused'))
2023-08-30 07:45:20,600 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 07:45:20,600 [DEBUG]  Node 'run_RegexMessageHandler' running 'RegexMessageHandler.process'.
2023-08-30 07:45:20,601 [DEBUG]  Received user message '1' with intent '{'name': 'ask_about_historicalFigure', 'confidence': 0.8700535297393799}' and entities '[{'entity': 'object', 'start': 0, 'end': 1, 'confidence_entity': 0.961410641670227, 'value': '1', 'extractor': 'DIETClassifier'}]'
2023-08-30 07:45:20,602 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 07:45:20,603 [DEBUG]  Logged UserUtterance - tracker now has 16 events.
2023-08-30 07:45:20,603 [DEBUG]  Validating extracted slots: 
2023-08-30 07:45:20,605 [DEBUG]  Default action 'action_extract_slots' was executed, resulting in 0 events: 
2023-08-30 07:45:20,607 [DEBUG]  Running graph with inputs: {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x7f7c5525a3a0>}, targets: ['select_prediction'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 07:45:20,608 [DEBUG]  Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
2023-08-30 07:45:20,609 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 07:45:20,610 [DEBUG]  Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
2023-08-30 07:45:20,612 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
2023-08-30 07:45:20,613 [DEBUG]  There is no memorised next action
2023-08-30 07:45:20,614 [DEBUG]  Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
2023-08-30 07:45:20,615 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 2] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 3] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 4] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 5] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
[state 6] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_one_condition
[state 7] user text: 1 | previous action name: action_listen
2023-08-30 07:45:20,615 [DEBUG]  There is no applicable rule.
2023-08-30 07:45:20,617 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 2] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 3] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 4] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 5] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
[state 6] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_one_condition
[state 7] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
2023-08-30 07:45:20,618 [DEBUG]  There is a rule for the next action 'action_one_condition'.
2023-08-30 07:45:20,619 [DEBUG]  Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
2023-08-30 07:45:20,625 [DEBUG]  TED predicted 'action_listen' based on user intent.
2023-08-30 07:45:20,627 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
2023-08-30 07:45:20,632 [DEBUG]  Querying for intent `ask_about_historicalFigure`.
2023-08-30 07:45:20,632 [DEBUG]  Query intent index 2 not found in label thresholds - {5: 1.5548791}. Check for `action_unlikely_intent` prediction will be skipped.
2023-08-30 07:45:20,633 [DEBUG]  Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
2023-08-30 07:45:20,633 [DEBUG]  Made prediction using user intent.
2023-08-30 07:45:20,634 [DEBUG]  Added `DefinePrevUserUtteredFeaturization(False)` event.
2023-08-30 07:45:20,635 [DEBUG]  Predicted next action using RulePolicy.
2023-08-30 07:45:20,636 [DEBUG]  Predicted next action 'action_one_condition' with confidence 1.00.
2023-08-30 07:45:20,637 [DEBUG]  Calling action endpoint to run action 'action_one_condition'.
2023-08-30 07:45:20,641 [ERROR]  Failed to run custom action 'action_one_condition'. Couldn't connect to the server at 'http://localhost:5055/webhook'. Is the server running? Error: Cannot connect to host localhost:5055 ssl:default [Connection refused]
2023-08-30 07:45:20,642 [ERROR]  Encountered an exception while running action 'action_one_condition'.Bot will continue, but the actions events are lost. Please check the logs of your action server for more information.
Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 980, in _wrap_create_connection
    return await self._loop.create_connection(*args, **kwargs)  # type: ignore[return-value]  # noqa
  File "uvloop/loop.pyx", line 2039, in create_connection
  File "uvloop/loop.pyx", line 2016, in uvloop.loop.Loop.create_connection
ConnectionRefusedError: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/actions/action.py", line 741, in run
    response: Any = await self.action_endpoint.request(
  File "/opt/venv/lib/python3.8/site-packages/rasa/utils/endpoints.py", line 164, in request
    async with session.request(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/client.py", line 1141, in __aenter__
    self._resp = await self._coro
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/client.py", line 536, in _request
    conn = await self._connector.connect(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 540, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 901, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 1206, in _create_direct_connection
    raise last_exc
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 1175, in _create_direct_connection
    transp, proto = await self._wrap_create_connection(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 988, in _wrap_create_connection
    raise client_error(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host localhost:5055 ssl:default [Connection refused]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/processor.py", line 874, in _run_action
    events = await action.run(
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/actions/action.py", line 776, in run
    raise RasaException(
rasa.shared.exceptions.RasaException: Failed to execute custom action 'action_one_condition'. Couldn't connect to the server at 'http://localhost:5055/webhook.
2023-08-30 07:45:20,643 [DEBUG]  Policy prediction ended with events '[<rasa.shared.core.events.DefinePrevUserUtteredFeaturization object at 0x7f7c5527b580>]'.
2023-08-30 07:45:20,644 [DEBUG]  Action 'action_one_condition' ended with events '[]'.
2023-08-30 07:45:20,645 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 07:45:20,646 [DEBUG]  Running graph with inputs: {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x7f7c5525a3a0>}, targets: ['select_prediction'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 07:45:20,647 [DEBUG]  Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
2023-08-30 07:45:20,648 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 07:45:20,649 [DEBUG]  Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
2023-08-30 07:45:20,650 [DEBUG]  Current tracker state:
[state 1] previous action name: action_listen
2023-08-30 07:45:20,651 [DEBUG]  There is no memorised next action
2023-08-30 07:45:20,652 [DEBUG]  Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
2023-08-30 07:45:20,653 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 2] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 3] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 4] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 5] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
[state 6] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_one_condition
[state 7] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
[state 8] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_one_condition
2023-08-30 07:45:20,654 [DEBUG]  There is a rule for the next action 'action_listen'.
2023-08-30 07:45:20,655 [DEBUG]  Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
2023-08-30 07:45:20,663 [DEBUG]  TED predicted 'action_listen' based on user intent.
2023-08-30 07:45:20,664 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
2023-08-30 07:45:20,664 [DEBUG]  Skipping predictions for UnexpecTEDIntentPolicy as either there is no event of type `UserUttered`, event's intent is new and not in domain or there is an event of type `ActionExecuted` after the last `UserUttered`.
2023-08-30 07:45:20,665 [DEBUG]  Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
2023-08-30 07:45:20,666 [DEBUG]  Predicted next action using RulePolicy.
2023-08-30 07:45:20,667 [DEBUG]  Predicted next action 'action_listen' with confidence 1.00.
2023-08-30 07:45:20,668 [DEBUG]  Policy prediction ended with events '[]'.
2023-08-30 07:45:20,668 [DEBUG]  Action 'action_listen' ended with events '[]'.
2023-08-30 07:45:20,669 [DEBUG]  Deleted lock for conversation 'user'.
2023-08-30 07:46:27,860 [DEBUG]  Issuing ticket for conversation 'user'.
2023-08-30 07:46:27,860 [DEBUG]  Acquiring lock for conversation 'user'.
2023-08-30 07:46:27,860 [DEBUG]  Acquired lock for conversation 'user'.
2023-08-30 07:46:27,862 [DEBUG]  Recreating tracker for id 'user'
2023-08-30 07:46:27,864 [DEBUG]  Running graph with inputs: {'__message__': [<rasa.core.channels.channel.UserMessage object at 0x7f7c5525a700>]}, targets: ['run_RegexMessageHandler'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 07:46:27,865 [DEBUG]  Node 'nlu_message_converter' running 'NLUMessageConverter.convert_user_message'.
2023-08-30 07:46:27,866 [DEBUG]  Node 'run_WhitespaceTokenizer0' running 'WhitespaceTokenizer.process'.
2023-08-30 07:46:27,867 [DEBUG]  Node 'run_RegexFeaturizer1' running 'RegexFeaturizer.process'.
2023-08-30 07:46:27,868 [DEBUG]  Node 'run_RegexEntityExtractor2' running 'RegexEntityExtractor.process'.
2023-08-30 07:46:27,869 [DEBUG]  Node 'run_LexicalSyntacticFeaturizer3' running 'LexicalSyntacticFeaturizer.process'.
2023-08-30 07:46:27,870 [DEBUG]  Node 'run_CountVectorsFeaturizer4' running 'CountVectorsFeaturizer.process'.
2023-08-30 07:46:27,871 [DEBUG]  Node 'run_CountVectorsFeaturizer5' running 'CountVectorsFeaturizer.process'.
2023-08-30 07:46:27,873 [DEBUG]  Node 'run_DIETClassifier6' running 'DIETClassifier.process'.
2023-08-30 07:46:27,885 [DEBUG]  Node 'run_EntitySynonymMapper7' running 'EntitySynonymMapper.process'.
2023-08-30 07:46:27,886 [DEBUG]  Node 'run_ResponseSelector8' running 'ResponseSelector.process'.
2023-08-30 07:46:27,886 [DEBUG]  There is no trained model for 'ResponseSelector': The component is either not trained or didn't receive enough training data.
2023-08-30 07:46:27,887 [DEBUG]  Adding following selector key to message property: default
2023-08-30 07:46:27,888 [DEBUG]  Node 'run_FallbackClassifier9' running 'FallbackClassifier.process'.
2023-08-30 07:46:27,889 [DEBUG]  Node 'run_DucklingEntityExtractor10' running 'DucklingEntityExtractor.process'.
2023-08-30 07:46:27,891 [DEBUG]  Starting new HTTP connection (1): localhost:8000
2023-08-30 07:46:27,892 [ERROR]  Failed to connect to duckling http server. Make sure the duckling server is running/healthy/not stale and the proper host and port are set in the configuration. More information on how to run the server can be found on github: https://github.com/facebook/duckling#quickstart Error: HTTPConnectionPool(host='localhost', port=8000): Max retries exceeded with url: /parse (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f7c551a1700>: Failed to establish a new connection: [Errno 111] Connection refused'))
2023-08-30 07:46:27,894 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 07:46:27,895 [DEBUG]  Node 'run_RegexMessageHandler' running 'RegexMessageHandler.process'.
2023-08-30 07:46:27,896 [DEBUG]  Received user message 'Who is Nguyen Ba Tinh ?' with intent '{'name': 'ask_about_historicalFigure', 'confidence': 1.0}' and entities '[{'entity': 'class', 'start': 0, 'end': 3, 'confidence_entity': 0.9999799728393555, 'value': 'culturaltourism:HistoricalFigure', 'extractor': 'DIETClassifier', 'processors': ['EntitySynonymMapper']}, {'entity': 'object', 'start': 7, 'end': 21, 'confidence_entity': 0.9999908208847046, 'value': 'culturaltourism:Nguyá»…n_BÃ¡_TÄ©nh', 'extractor': 'DIETClassifier', 'processors': ['EntitySynonymMapper']}]'
2023-08-30 07:46:27,896 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 07:46:27,897 [DEBUG]  Logged UserUtterance - tracker now has 20 events.
2023-08-30 07:46:27,898 [DEBUG]  Validating extracted slots: 
2023-08-30 07:46:27,898 [DEBUG]  Default action 'action_extract_slots' was executed, resulting in 0 events: 
2023-08-30 07:46:27,899 [DEBUG]  Running graph with inputs: {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x7f7c5527b6d0>}, targets: ['select_prediction'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 07:46:27,900 [DEBUG]  Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
2023-08-30 07:46:27,902 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 07:46:27,903 [DEBUG]  Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
2023-08-30 07:46:27,904 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
2023-08-30 07:46:27,905 [DEBUG]  There is no memorised next action
2023-08-30 07:46:27,905 [DEBUG]  Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
2023-08-30 07:46:27,907 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 2] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 3] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 4] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 5] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
[state 6] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_one_condition
[state 7] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
[state 8] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_one_condition
[state 9] user text: Who is Nguyen Ba Tinh ? | previous action name: action_listen
2023-08-30 07:46:27,907 [DEBUG]  There is no applicable rule.
2023-08-30 07:46:27,908 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 2] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 3] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 4] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 5] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
[state 6] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_one_condition
[state 7] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
[state 8] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_one_condition
[state 9] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
2023-08-30 07:46:27,909 [DEBUG]  There is a rule for the next action 'action_one_condition'.
2023-08-30 07:46:27,910 [DEBUG]  Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
2023-08-30 07:46:27,916 [DEBUG]  TED predicted 'action_listing' based on user intent.
2023-08-30 07:46:27,917 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
2023-08-30 07:46:27,922 [DEBUG]  Querying for intent `ask_about_historicalFigure`.
2023-08-30 07:46:27,923 [DEBUG]  Query intent index 2 not found in label thresholds - {5: 1.5548791}. Check for `action_unlikely_intent` prediction will be skipped.
2023-08-30 07:46:27,923 [DEBUG]  Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
2023-08-30 07:46:27,924 [DEBUG]  Made prediction using user intent.
2023-08-30 07:46:27,925 [DEBUG]  Added `DefinePrevUserUtteredFeaturization(False)` event.
2023-08-30 07:46:27,925 [DEBUG]  Predicted next action using RulePolicy.
2023-08-30 07:46:27,927 [DEBUG]  Predicted next action 'action_one_condition' with confidence 1.00.
2023-08-30 07:46:27,928 [DEBUG]  Calling action endpoint to run action 'action_one_condition'.
2023-08-30 07:46:27,932 [ERROR]  Failed to run custom action 'action_one_condition'. Couldn't connect to the server at 'http://localhost:5055/webhook'. Is the server running? Error: Cannot connect to host localhost:5055 ssl:default [Connection refused]
2023-08-30 07:46:27,932 [ERROR]  Encountered an exception while running action 'action_one_condition'.Bot will continue, but the actions events are lost. Please check the logs of your action server for more information.
Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 980, in _wrap_create_connection
    return await self._loop.create_connection(*args, **kwargs)  # type: ignore[return-value]  # noqa
  File "uvloop/loop.pyx", line 2039, in create_connection
  File "uvloop/loop.pyx", line 2016, in uvloop.loop.Loop.create_connection
ConnectionRefusedError: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/actions/action.py", line 741, in run
    response: Any = await self.action_endpoint.request(
  File "/opt/venv/lib/python3.8/site-packages/rasa/utils/endpoints.py", line 164, in request
    async with session.request(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/client.py", line 1141, in __aenter__
    self._resp = await self._coro
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/client.py", line 536, in _request
    conn = await self._connector.connect(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 540, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 901, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 1206, in _create_direct_connection
    raise last_exc
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 1175, in _create_direct_connection
    transp, proto = await self._wrap_create_connection(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 988, in _wrap_create_connection
    raise client_error(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host localhost:5055 ssl:default [Connection refused]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/processor.py", line 874, in _run_action
    events = await action.run(
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/actions/action.py", line 776, in run
    raise RasaException(
rasa.shared.exceptions.RasaException: Failed to execute custom action 'action_one_condition'. Couldn't connect to the server at 'http://localhost:5055/webhook.
2023-08-30 07:46:27,934 [DEBUG]  Policy prediction ended with events '[<rasa.shared.core.events.DefinePrevUserUtteredFeaturization object at 0x7f7c551a13a0>]'.
2023-08-30 07:46:27,935 [DEBUG]  Action 'action_one_condition' ended with events '[]'.
2023-08-30 07:46:27,937 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 07:46:27,938 [DEBUG]  Running graph with inputs: {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x7f7c5527b6d0>}, targets: ['select_prediction'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 07:46:27,943 [DEBUG]  Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
2023-08-30 07:46:27,945 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 07:46:27,945 [DEBUG]  Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
2023-08-30 07:46:27,946 [DEBUG]  Current tracker state:
[state 1] previous action name: action_listen
2023-08-30 07:46:27,946 [DEBUG]  There is no memorised next action
2023-08-30 07:46:27,947 [DEBUG]  Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
2023-08-30 07:46:27,949 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 2] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 3] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 4] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
[state 5] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
[state 6] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_one_condition
[state 7] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
[state 8] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_one_condition
[state 9] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 10] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
2023-08-30 07:46:27,950 [DEBUG]  There is a rule for the next action 'action_listen'.
2023-08-30 07:46:27,952 [DEBUG]  Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
2023-08-30 07:46:27,959 [DEBUG]  TED predicted 'action_listen' based on user intent.
2023-08-30 07:46:27,959 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
2023-08-30 07:46:27,960 [DEBUG]  Skipping predictions for UnexpecTEDIntentPolicy as either there is no event of type `UserUttered`, event's intent is new and not in domain or there is an event of type `ActionExecuted` after the last `UserUttered`.
2023-08-30 07:46:27,960 [DEBUG]  Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
2023-08-30 07:46:27,961 [DEBUG]  Predicted next action using RulePolicy.
2023-08-30 07:46:27,962 [DEBUG]  Predicted next action 'action_listen' with confidence 1.00.
2023-08-30 07:46:27,963 [DEBUG]  Policy prediction ended with events '[]'.
2023-08-30 07:46:27,963 [DEBUG]  Action 'action_listen' ended with events '[]'.
2023-08-30 07:46:27,964 [DEBUG]  Deleted lock for conversation 'user'.
2023-08-30 07:58:38,374 [DEBUG]  Available web server routes: 
/conversations/<conversation_id:path>/messages     POST                           add_message
/conversations/<conversation_id:path>/tracker/events POST                           append_events
/webhooks/rasa                                     GET                            custom_webhook_RasaChatInput.health
/webhooks/rasa/webhook                             POST                           custom_webhook_RasaChatInput.receive
/webhooks/rest                                     GET                            custom_webhook_RestInput.health
/webhooks/rest/webhook                             POST                           custom_webhook_RestInput.receive
/model/test/intents                                POST                           evaluate_intents
/model/test/stories                                POST                           evaluate_stories
/conversations/<conversation_id:path>/execute      POST                           execute_action
/domain                                            GET                            get_domain
/                                                  GET                            hello
/model                                             PUT                            load_model
/model/parse                                       POST                           parse
/conversations/<conversation_id:path>/predict      POST                           predict
/conversations/<conversation_id:path>/tracker/events PUT                            replace_events
/conversations/<conversation_id:path>/story        GET                            retrieve_story
/conversations/<conversation_id:path>/tracker      GET                            retrieve_tracker
/status                                            GET                            status
/model/predict                                     POST                           tracker_predict
/model/train                                       POST                           train
/conversations/<conversation_id:path>/trigger_intent POST                           trigger_intent
/model                                             DELETE                         unload_model
/version                                           GET                            version
2023-08-30 07:58:38,389 [INFO ]  Starting Rasa server on http://0.0.0.0:5005
2023-08-30 07:58:38,390 [DEBUG]  Using the default number of Sanic workers (1).
2023-08-30 07:58:39,147 [DEBUG]  Starting new HTTPS connection (1): api.segment.io:443
2023-08-30 07:58:39,995 [DEBUG]  https://api.segment.io:443 "POST /v1/track HTTP/1.1" 200 21
2023-08-30 07:58:40,042 [DEBUG]  Connected to InMemoryTrackerStore.
2023-08-30 07:58:40,043 [DEBUG]  Connected to lock store 'InMemoryLockStore'.
2023-08-30 07:58:40,043 [DEBUG]  Instantiated NLG to 'TemplatedNaturalLanguageGenerator'.
2023-08-30 07:58:40,044 [INFO ]  Loading model /app/models/20230830-073139-augmenting-convolution.tar.gz...
2023-08-30 07:58:40,670 [DEBUG]  Extracted model to '/tmp/tmp0_egdxod'.
2023-08-30 07:58:40,673 [DEBUG]  Node 'nlu_message_converter' loading 'NLUMessageConverter.load' and kwargs: '{}'.
2023-08-30 07:58:40,673 [DEBUG]  Node 'run_WhitespaceTokenizer0' loading 'WhitespaceTokenizer.load' and kwargs: '{}'.
2023-08-30 07:58:40,676 [DEBUG]  Node 'run_RegexFeaturizer1' loading 'RegexFeaturizer.load' and kwargs: '{}'.
2023-08-30 07:58:40,676 [DEBUG]  Resource 'train_RegexFeaturizer1' was requested for reading.
2023-08-30 07:58:40,676 [DEBUG]  Node 'run_RegexEntityExtractor2' loading 'RegexEntityExtractor.load' and kwargs: '{}'.
2023-08-30 07:58:40,677 [DEBUG]  Resource 'train_RegexEntityExtractor2' was requested for reading.
2023-08-30 07:58:40,677 [DEBUG]  Node 'run_LexicalSyntacticFeaturizer3' loading 'LexicalSyntacticFeaturizer.load' and kwargs: '{}'.
2023-08-30 07:58:40,678 [DEBUG]  Resource 'train_LexicalSyntacticFeaturizer3' was requested for reading.
2023-08-30 07:58:40,679 [DEBUG]  Node 'run_CountVectorsFeaturizer4' loading 'CountVectorsFeaturizer.load' and kwargs: '{}'.
2023-08-30 07:58:40,679 [DEBUG]  Resource 'train_CountVectorsFeaturizer4' was requested for reading.
2023-08-30 07:58:40,682 [DEBUG]  Node 'run_CountVectorsFeaturizer5' loading 'CountVectorsFeaturizer.load' and kwargs: '{}'.
2023-08-30 07:58:40,682 [DEBUG]  Resource 'train_CountVectorsFeaturizer5' was requested for reading.
2023-08-30 07:58:40,691 [DEBUG]  Node 'run_DIETClassifier6' loading 'DIETClassifier.load' and kwargs: '{}'.
2023-08-30 07:58:40,691 [DEBUG]  Resource 'train_DIETClassifier6' was requested for reading.
2023-08-30 07:58:40,692 [DEBUG]  Loading the model from /tmp/tmp2swoi9y9/train_DIETClassifier6/DIETClassifier.tf_model with finetune_mode=False...
2023-08-30 07:58:40,735 [DEBUG]  Following metrics will be logged during training: 
2023-08-30 07:58:40,735 [DEBUG]    t_loss (total loss)
2023-08-30 07:58:40,736 [DEBUG]    i_acc (intent acc)
2023-08-30 07:58:40,736 [DEBUG]    i_loss (intent loss)
2023-08-30 07:58:40,737 [DEBUG]    e_f1 (entity f1)
2023-08-30 07:58:40,737 [DEBUG]    e_loss (entity loss)
2023-08-30 07:59:09,770 [DEBUG]  Finished loading the model.
2023-08-30 07:59:09,771 [DEBUG]  Node 'run_EntitySynonymMapper7' loading 'EntitySynonymMapper.load' and kwargs: '{}'.
2023-08-30 07:59:09,771 [DEBUG]  Resource 'train_EntitySynonymMapper7' was requested for reading.
2023-08-30 07:59:09,772 [DEBUG]  Node 'run_ResponseSelector8' loading 'ResponseSelector.load' and kwargs: '{}'.
2023-08-30 07:59:09,773 [DEBUG]  Resource 'train_ResponseSelector8' was requested for reading.
2023-08-30 07:59:09,773 [DEBUG]  Failed to load ABCMeta from model storage. Resource 'train_ResponseSelector8' doesn't exist.
2023-08-30 07:59:09,774 [DEBUG]  Resource 'train_ResponseSelector8' was requested for reading.
2023-08-30 07:59:09,774 [DEBUG]  Failed to load ResponseSelector from model storage. Resource 'train_ResponseSelector8' doesn't exist.
2023-08-30 07:59:09,774 [DEBUG]  Node 'run_FallbackClassifier9' loading 'FallbackClassifier.load' and kwargs: '{}'.
2023-08-30 07:59:09,775 [DEBUG]  Node 'run_DucklingEntityExtractor10' loading 'DucklingEntityExtractor.load' and kwargs: '{}'.
2023-08-30 07:59:09,775 [DEBUG]  Node 'run_RegexMessageHandler' loading 'RegexMessageHandler.load' and kwargs: '{}'.
2023-08-30 07:59:09,775 [DEBUG]  Node 'domain_provider' loading 'DomainProvider.load' and kwargs: '{}'.
2023-08-30 07:59:09,776 [DEBUG]  Resource 'domain_provider' was requested for reading.
2023-08-30 07:59:09,789 [DEBUG]  Node 'run_MemoizationPolicy0' loading 'MemoizationPolicy.load' and kwargs: '{}'.
2023-08-30 07:59:09,789 [DEBUG]  Resource 'train_MemoizationPolicy0' was requested for reading.
2023-08-30 07:59:09,790 [DEBUG]  Node 'run_RulePolicy1' loading 'RulePolicy.load' and kwargs: '{}'.
2023-08-30 07:59:09,791 [DEBUG]  Resource 'train_RulePolicy1' was requested for reading.
2023-08-30 07:59:09,791 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' loading 'UnexpecTEDIntentPolicy.load' and kwargs: '{}'.
2023-08-30 07:59:09,792 [DEBUG]  Resource 'train_UnexpecTEDIntentPolicy2' was requested for reading.
2023-08-30 07:59:09,793 [DEBUG]  Loading the model from /tmp/tmp2swoi9y9/train_UnexpecTEDIntentPolicy2/unexpected_intent_policy.tf_model with finetune_mode=False...
2023-08-30 07:59:19,969 [DEBUG]  Finished loading the model.
2023-08-30 07:59:19,970 [WARNI]  The UnexpecTED Intent Policy is currently experimental and might change or be removed in the future ðŸ”¬ Please share your feedback on it in the forum (https://forum.rasa.com) to help us make this feature ready for production.
2023-08-30 07:59:19,970 [DEBUG]  Node 'run_TEDPolicy3' loading 'TEDPolicy.load' and kwargs: '{}'.
2023-08-30 07:59:19,971 [DEBUG]  Resource 'train_TEDPolicy3' was requested for reading.
2023-08-30 07:59:19,972 [DEBUG]  Loading the model from /tmp/tmp2swoi9y9/train_TEDPolicy3/ted_policy.tf_model with finetune_mode=False...
2023-08-30 07:59:29,631 [DEBUG]  Finished loading the model.
2023-08-30 07:59:29,632 [DEBUG]  Node 'rule_only_data_provider' loading 'RuleOnlyDataProvider.load' and kwargs: '{}'.
2023-08-30 07:59:29,632 [DEBUG]  Resource 'train_RulePolicy1' was requested for reading.
2023-08-30 07:59:29,632 [DEBUG]  Node 'select_prediction' loading 'DefaultPolicyPredictionEnsemble.load' and kwargs: '{}'.
2023-08-30 07:59:29,639 [INFO ]  Rasa server is up and running.
2023-08-30 07:59:29,640 [INFO ]  Enabling coroutine debugging. Loop id 106415296.
2023-08-30 07:59:40,770 [DEBUG]  Available web server routes: 
/conversations/<conversation_id:path>/messages     POST                           add_message
/conversations/<conversation_id:path>/tracker/events POST                           append_events
/webhooks/rasa                                     GET                            custom_webhook_RasaChatInput.health
/webhooks/rasa/webhook                             POST                           custom_webhook_RasaChatInput.receive
/webhooks/rest                                     GET                            custom_webhook_RestInput.health
/webhooks/rest/webhook                             POST                           custom_webhook_RestInput.receive
/model/test/intents                                POST                           evaluate_intents
/model/test/stories                                POST                           evaluate_stories
/conversations/<conversation_id:path>/execute      POST                           execute_action
/domain                                            GET                            get_domain
/                                                  GET                            hello
/model                                             PUT                            load_model
/model/parse                                       POST                           parse
/conversations/<conversation_id:path>/predict      POST                           predict
/conversations/<conversation_id:path>/tracker/events PUT                            replace_events
/conversations/<conversation_id:path>/story        GET                            retrieve_story
/conversations/<conversation_id:path>/tracker      GET                            retrieve_tracker
/status                                            GET                            status
/model/predict                                     POST                           tracker_predict
/model/train                                       POST                           train
/conversations/<conversation_id:path>/trigger_intent POST                           trigger_intent
/model                                             DELETE                         unload_model
/version                                           GET                            version
2023-08-30 07:59:40,773 [INFO ]  Starting Rasa server on http://0.0.0.0:5005
2023-08-30 07:59:40,773 [DEBUG]  Using the default number of Sanic workers (1).
2023-08-30 07:59:41,495 [DEBUG]  Starting new HTTPS connection (1): api.segment.io:443
2023-08-30 07:59:42,366 [DEBUG]  https://api.segment.io:443 "POST /v1/track HTTP/1.1" 200 21
2023-08-30 07:59:42,409 [DEBUG]  Connected to InMemoryTrackerStore.
2023-08-30 07:59:42,410 [DEBUG]  Connected to lock store 'InMemoryLockStore'.
2023-08-30 07:59:42,410 [DEBUG]  Instantiated NLG to 'TemplatedNaturalLanguageGenerator'.
2023-08-30 07:59:42,411 [INFO ]  Loading model /app/models/20230830-073139-augmenting-convolution.tar.gz...
2023-08-30 07:59:43,049 [DEBUG]  Extracted model to '/tmp/tmpeklv34qm'.
2023-08-30 07:59:43,051 [DEBUG]  Node 'nlu_message_converter' loading 'NLUMessageConverter.load' and kwargs: '{}'.
2023-08-30 07:59:43,051 [DEBUG]  Node 'run_WhitespaceTokenizer0' loading 'WhitespaceTokenizer.load' and kwargs: '{}'.
2023-08-30 07:59:43,054 [DEBUG]  Node 'run_RegexFeaturizer1' loading 'RegexFeaturizer.load' and kwargs: '{}'.
2023-08-30 07:59:43,055 [DEBUG]  Resource 'train_RegexFeaturizer1' was requested for reading.
2023-08-30 07:59:43,055 [DEBUG]  Node 'run_RegexEntityExtractor2' loading 'RegexEntityExtractor.load' and kwargs: '{}'.
2023-08-30 07:59:43,055 [DEBUG]  Resource 'train_RegexEntityExtractor2' was requested for reading.
2023-08-30 07:59:43,056 [DEBUG]  Node 'run_LexicalSyntacticFeaturizer3' loading 'LexicalSyntacticFeaturizer.load' and kwargs: '{}'.
2023-08-30 07:59:43,056 [DEBUG]  Resource 'train_LexicalSyntacticFeaturizer3' was requested for reading.
2023-08-30 07:59:43,058 [DEBUG]  Node 'run_CountVectorsFeaturizer4' loading 'CountVectorsFeaturizer.load' and kwargs: '{}'.
2023-08-30 07:59:43,058 [DEBUG]  Resource 'train_CountVectorsFeaturizer4' was requested for reading.
2023-08-30 07:59:43,061 [DEBUG]  Node 'run_CountVectorsFeaturizer5' loading 'CountVectorsFeaturizer.load' and kwargs: '{}'.
2023-08-30 07:59:43,061 [DEBUG]  Resource 'train_CountVectorsFeaturizer5' was requested for reading.
2023-08-30 07:59:43,070 [DEBUG]  Node 'run_DIETClassifier6' loading 'DIETClassifier.load' and kwargs: '{}'.
2023-08-30 07:59:43,070 [DEBUG]  Resource 'train_DIETClassifier6' was requested for reading.
2023-08-30 07:59:43,071 [DEBUG]  Loading the model from /tmp/tmpgy2ent7c/train_DIETClassifier6/DIETClassifier.tf_model with finetune_mode=False...
2023-08-30 07:59:43,118 [DEBUG]  Following metrics will be logged during training: 
2023-08-30 07:59:43,118 [DEBUG]    t_loss (total loss)
2023-08-30 07:59:43,119 [DEBUG]    i_acc (intent acc)
2023-08-30 07:59:43,119 [DEBUG]    i_loss (intent loss)
2023-08-30 07:59:43,119 [DEBUG]    e_f1 (entity f1)
2023-08-30 07:59:43,120 [DEBUG]    e_loss (entity loss)
2023-08-30 08:00:11,896 [DEBUG]  Finished loading the model.
2023-08-30 08:00:11,896 [DEBUG]  Node 'run_EntitySynonymMapper7' loading 'EntitySynonymMapper.load' and kwargs: '{}'.
2023-08-30 08:00:11,897 [DEBUG]  Resource 'train_EntitySynonymMapper7' was requested for reading.
2023-08-30 08:00:11,898 [DEBUG]  Node 'run_ResponseSelector8' loading 'ResponseSelector.load' and kwargs: '{}'.
2023-08-30 08:00:11,899 [DEBUG]  Resource 'train_ResponseSelector8' was requested for reading.
2023-08-30 08:00:11,900 [DEBUG]  Failed to load ABCMeta from model storage. Resource 'train_ResponseSelector8' doesn't exist.
2023-08-30 08:00:11,900 [DEBUG]  Resource 'train_ResponseSelector8' was requested for reading.
2023-08-30 08:00:11,901 [DEBUG]  Failed to load ResponseSelector from model storage. Resource 'train_ResponseSelector8' doesn't exist.
2023-08-30 08:00:11,902 [DEBUG]  Node 'run_FallbackClassifier9' loading 'FallbackClassifier.load' and kwargs: '{}'.
2023-08-30 08:00:11,902 [DEBUG]  Node 'run_DucklingEntityExtractor10' loading 'DucklingEntityExtractor.load' and kwargs: '{}'.
2023-08-30 08:00:11,903 [DEBUG]  Node 'run_RegexMessageHandler' loading 'RegexMessageHandler.load' and kwargs: '{}'.
2023-08-30 08:00:11,904 [DEBUG]  Node 'domain_provider' loading 'DomainProvider.load' and kwargs: '{}'.
2023-08-30 08:00:11,904 [DEBUG]  Resource 'domain_provider' was requested for reading.
2023-08-30 08:00:11,919 [DEBUG]  Node 'run_MemoizationPolicy0' loading 'MemoizationPolicy.load' and kwargs: '{}'.
2023-08-30 08:00:11,920 [DEBUG]  Resource 'train_MemoizationPolicy0' was requested for reading.
2023-08-30 08:00:11,921 [DEBUG]  Node 'run_RulePolicy1' loading 'RulePolicy.load' and kwargs: '{}'.
2023-08-30 08:00:11,921 [DEBUG]  Resource 'train_RulePolicy1' was requested for reading.
2023-08-30 08:00:11,922 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' loading 'UnexpecTEDIntentPolicy.load' and kwargs: '{}'.
2023-08-30 08:00:11,922 [DEBUG]  Resource 'train_UnexpecTEDIntentPolicy2' was requested for reading.
2023-08-30 08:00:11,924 [DEBUG]  Loading the model from /tmp/tmpgy2ent7c/train_UnexpecTEDIntentPolicy2/unexpected_intent_policy.tf_model with finetune_mode=False...
2023-08-30 08:00:22,129 [DEBUG]  Finished loading the model.
2023-08-30 08:00:22,130 [WARNI]  The UnexpecTED Intent Policy is currently experimental and might change or be removed in the future ðŸ”¬ Please share your feedback on it in the forum (https://forum.rasa.com) to help us make this feature ready for production.
2023-08-30 08:00:22,131 [DEBUG]  Node 'run_TEDPolicy3' loading 'TEDPolicy.load' and kwargs: '{}'.
2023-08-30 08:00:22,131 [DEBUG]  Resource 'train_TEDPolicy3' was requested for reading.
2023-08-30 08:00:22,132 [DEBUG]  Loading the model from /tmp/tmpgy2ent7c/train_TEDPolicy3/ted_policy.tf_model with finetune_mode=False...
2023-08-30 08:00:31,489 [DEBUG]  Finished loading the model.
2023-08-30 08:00:31,489 [DEBUG]  Node 'rule_only_data_provider' loading 'RuleOnlyDataProvider.load' and kwargs: '{}'.
2023-08-30 08:00:31,490 [DEBUG]  Resource 'train_RulePolicy1' was requested for reading.
2023-08-30 08:00:31,490 [DEBUG]  Node 'select_prediction' loading 'DefaultPolicyPredictionEnsemble.load' and kwargs: '{}'.
2023-08-30 08:00:31,497 [INFO ]  Rasa server is up and running.
2023-08-30 08:00:31,499 [INFO ]  Enabling coroutine debugging. Loop id 113113136.
2023-08-30 08:00:38,046 [DEBUG]  Issuing ticket for conversation 'user'.
2023-08-30 08:00:38,046 [DEBUG]  Acquiring lock for conversation 'user'.
2023-08-30 08:00:38,046 [DEBUG]  Acquired lock for conversation 'user'.
2023-08-30 08:00:38,047 [DEBUG]  Could not find tracker for conversation ID 'user'.
2023-08-30 08:00:38,047 [DEBUG]  Starting a new session for conversation ID 'user'.
2023-08-30 08:00:38,049 [DEBUG]  Policy prediction ended with events '[]'.
2023-08-30 08:00:38,049 [DEBUG]  Action 'action_session_start' ended with events '[<rasa.shared.core.events.SessionStarted object at 0x7ff142e65e50>, ActionExecuted(action: action_listen, policy: None, confidence: None)]'.
2023-08-30 08:00:38,050 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 08:00:38,050 [DEBUG]  Running graph with inputs: {'__message__': [<rasa.core.channels.channel.UserMessage object at 0x7ff142e65c70>]}, targets: ['run_RegexMessageHandler'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 08:00:38,051 [DEBUG]  Node 'nlu_message_converter' running 'NLUMessageConverter.convert_user_message'.
2023-08-30 08:00:38,051 [DEBUG]  Node 'run_WhitespaceTokenizer0' running 'WhitespaceTokenizer.process'.
2023-08-30 08:00:38,053 [DEBUG]  Node 'run_RegexFeaturizer1' running 'RegexFeaturizer.process'.
2023-08-30 08:00:38,055 [DEBUG]  Node 'run_RegexEntityExtractor2' running 'RegexEntityExtractor.process'.
2023-08-30 08:00:38,057 [DEBUG]  Node 'run_LexicalSyntacticFeaturizer3' running 'LexicalSyntacticFeaturizer.process'.
2023-08-30 08:00:38,058 [DEBUG]  Node 'run_CountVectorsFeaturizer4' running 'CountVectorsFeaturizer.process'.
2023-08-30 08:00:38,060 [DEBUG]  Node 'run_CountVectorsFeaturizer5' running 'CountVectorsFeaturizer.process'.
2023-08-30 08:00:38,061 [DEBUG]  Node 'run_DIETClassifier6' running 'DIETClassifier.process'.
2023-08-30 08:00:38,387 [DEBUG]  Node 'run_EntitySynonymMapper7' running 'EntitySynonymMapper.process'.
2023-08-30 08:00:38,387 [DEBUG]  Node 'run_ResponseSelector8' running 'ResponseSelector.process'.
2023-08-30 08:00:38,388 [DEBUG]  There is no trained model for 'ResponseSelector': The component is either not trained or didn't receive enough training data.
2023-08-30 08:00:38,389 [DEBUG]  Adding following selector key to message property: default
2023-08-30 08:00:38,391 [DEBUG]  Node 'run_FallbackClassifier9' running 'FallbackClassifier.process'.
2023-08-30 08:00:38,393 [DEBUG]  Node 'run_DucklingEntityExtractor10' running 'DucklingEntityExtractor.process'.
2023-08-30 08:00:38,395 [DEBUG]  Starting new HTTP connection (1): localhost:8000
2023-08-30 08:00:38,397 [ERROR]  Failed to connect to duckling http server. Make sure the duckling server is running/healthy/not stale and the proper host and port are set in the configuration. More information on how to run the server can be found on github: https://github.com/facebook/duckling#quickstart Error: HTTPConnectionPool(host='localhost', port=8000): Max retries exceeded with url: /parse (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff142dd70d0>: Failed to establish a new connection: [Errno 111] Connection refused'))
2023-08-30 08:00:38,397 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 08:00:38,398 [DEBUG]  Node 'run_RegexMessageHandler' running 'RegexMessageHandler.process'.
2023-08-30 08:00:38,398 [DEBUG]  Received user message 'Who is Nguyen Ba Tinh ?' with intent '{'name': 'ask_about_historicalFigure', 'confidence': 1.0}' and entities '[{'entity': 'class', 'start': 0, 'end': 3, 'confidence_entity': 0.9999799728393555, 'value': 'culturaltourism:HistoricalFigure', 'extractor': 'DIETClassifier', 'processors': ['EntitySynonymMapper']}, {'entity': 'object', 'start': 7, 'end': 21, 'confidence_entity': 0.9999908208847046, 'value': 'culturaltourism:Nguyá»…n_BÃ¡_TÄ©nh', 'extractor': 'DIETClassifier', 'processors': ['EntitySynonymMapper']}]'
2023-08-30 08:00:38,399 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 08:00:38,400 [DEBUG]  Logged UserUtterance - tracker now has 4 events.
2023-08-30 08:00:38,400 [DEBUG]  Validating extracted slots: 
2023-08-30 08:00:38,400 [DEBUG]  Default action 'action_extract_slots' was executed, resulting in 0 events: 
2023-08-30 08:00:38,401 [DEBUG]  Running graph with inputs: {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x7ff142e65d30>}, targets: ['select_prediction'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 08:00:38,402 [DEBUG]  Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
2023-08-30 08:00:38,403 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 08:00:38,403 [DEBUG]  Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
2023-08-30 08:00:38,404 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
2023-08-30 08:00:38,404 [DEBUG]  There is no memorised next action
2023-08-30 08:00:38,405 [DEBUG]  Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
2023-08-30 08:00:38,406 [DEBUG]  Current tracker state:
[state 1] user text: Who is Nguyen Ba Tinh ? | previous action name: action_listen
2023-08-30 08:00:38,408 [DEBUG]  There is no applicable rule.
2023-08-30 08:00:38,408 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
2023-08-30 08:00:38,409 [DEBUG]  There is a rule for the next action 'action_one_condition'.
2023-08-30 08:00:38,409 [DEBUG]  Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
2023-08-30 08:00:38,415 [DEBUG]  TED predicted 'action_listing' based on user intent.
2023-08-30 08:00:38,415 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
2023-08-30 08:00:38,770 [DEBUG]  Querying for intent `ask_about_historicalFigure`.
2023-08-30 08:00:38,770 [DEBUG]  Query intent index 2 not found in label thresholds - {5: 1.5548791}. Check for `action_unlikely_intent` prediction will be skipped.
2023-08-30 08:00:38,771 [DEBUG]  Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
2023-08-30 08:00:38,771 [DEBUG]  Made prediction using user intent.
2023-08-30 08:00:38,772 [DEBUG]  Added `DefinePrevUserUtteredFeaturization(False)` event.
2023-08-30 08:00:38,772 [DEBUG]  Predicted next action using RulePolicy.
2023-08-30 08:00:38,773 [DEBUG]  Predicted next action 'action_one_condition' with confidence 1.00.
2023-08-30 08:00:38,773 [DEBUG]  Calling action endpoint to run action 'action_one_condition'.
2023-08-30 08:00:38,779 [ERROR]  Failed to run custom action 'action_one_condition'. Couldn't connect to the server at 'http://localhost:5055/webhook'. Is the server running? Error: Cannot connect to host localhost:5055 ssl:default [Connection refused]
2023-08-30 08:00:38,780 [ERROR]  Encountered an exception while running action 'action_one_condition'.Bot will continue, but the actions events are lost. Please check the logs of your action server for more information.
Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 980, in _wrap_create_connection
    return await self._loop.create_connection(*args, **kwargs)  # type: ignore[return-value]  # noqa
  File "uvloop/loop.pyx", line 2039, in create_connection
  File "uvloop/loop.pyx", line 2016, in uvloop.loop.Loop.create_connection
ConnectionRefusedError: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/actions/action.py", line 741, in run
    response: Any = await self.action_endpoint.request(
  File "/opt/venv/lib/python3.8/site-packages/rasa/utils/endpoints.py", line 164, in request
    async with session.request(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/client.py", line 1141, in __aenter__
    self._resp = await self._coro
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/client.py", line 536, in _request
    conn = await self._connector.connect(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 540, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 901, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 1206, in _create_direct_connection
    raise last_exc
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 1175, in _create_direct_connection
    transp, proto = await self._wrap_create_connection(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 988, in _wrap_create_connection
    raise client_error(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host localhost:5055 ssl:default [Connection refused]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/processor.py", line 874, in _run_action
    events = await action.run(
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/actions/action.py", line 776, in run
    raise RasaException(
rasa.shared.exceptions.RasaException: Failed to execute custom action 'action_one_condition'. Couldn't connect to the server at 'http://localhost:5055/webhook.
2023-08-30 08:00:38,781 [DEBUG]  Policy prediction ended with events '[<rasa.shared.core.events.DefinePrevUserUtteredFeaturization object at 0x7ff142e35d30>]'.
2023-08-30 08:00:38,782 [DEBUG]  Action 'action_one_condition' ended with events '[]'.
2023-08-30 08:00:38,782 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 08:00:38,782 [DEBUG]  Running graph with inputs: {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x7ff142e65d30>}, targets: ['select_prediction'] and ExecutionContext(model_id='f50ffb52b0424f2a8c95e93d58f5bd95', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 08:00:38,783 [DEBUG]  Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
2023-08-30 08:00:38,783 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 08:00:38,784 [DEBUG]  Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
2023-08-30 08:00:38,784 [DEBUG]  Current tracker state:
[state 1] previous action name: action_listen
2023-08-30 08:00:38,785 [DEBUG]  There is no memorised next action
2023-08-30 08:00:38,785 [DEBUG]  Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
2023-08-30 08:00:38,786 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_listen
[state 2] user intent: ask_about_historicalFigure | user entities: ('class', 'object') | previous action name: action_one_condition
2023-08-30 08:00:38,786 [DEBUG]  There is a rule for the next action 'action_listen'.
2023-08-30 08:00:38,786 [DEBUG]  Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
2023-08-30 08:00:38,791 [DEBUG]  TED predicted 'action_listen' based on user intent.
2023-08-30 08:00:38,792 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
2023-08-30 08:00:38,792 [DEBUG]  Skipping predictions for UnexpecTEDIntentPolicy as either there is no event of type `UserUttered`, event's intent is new and not in domain or there is an event of type `ActionExecuted` after the last `UserUttered`.
2023-08-30 08:00:38,793 [DEBUG]  Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
2023-08-30 08:00:38,793 [DEBUG]  Predicted next action using RulePolicy.
2023-08-30 08:00:38,794 [DEBUG]  Predicted next action 'action_listen' with confidence 1.00.
2023-08-30 08:00:38,795 [DEBUG]  Policy prediction ended with events '[]'.
2023-08-30 08:00:38,795 [DEBUG]  Action 'action_listen' ended with events '[]'.
2023-08-30 08:00:38,795 [DEBUG]  Deleted lock for conversation 'user'.
2023-08-30 08:19:57,305 [DEBUG]  Available web server routes: 
/conversations/<conversation_id:path>/messages     POST                           add_message
/conversations/<conversation_id:path>/tracker/events POST                           append_events
/webhooks/rasa                                     GET                            custom_webhook_RasaChatInput.health
/webhooks/rasa/webhook                             POST                           custom_webhook_RasaChatInput.receive
/webhooks/rest                                     GET                            custom_webhook_RestInput.health
/webhooks/rest/webhook                             POST                           custom_webhook_RestInput.receive
/model/test/intents                                POST                           evaluate_intents
/model/test/stories                                POST                           evaluate_stories
/conversations/<conversation_id:path>/execute      POST                           execute_action
/domain                                            GET                            get_domain
/                                                  GET                            hello
/model                                             PUT                            load_model
/model/parse                                       POST                           parse
/conversations/<conversation_id:path>/predict      POST                           predict
/conversations/<conversation_id:path>/tracker/events PUT                            replace_events
/conversations/<conversation_id:path>/story        GET                            retrieve_story
/conversations/<conversation_id:path>/tracker      GET                            retrieve_tracker
/status                                            GET                            status
/model/predict                                     POST                           tracker_predict
/model/train                                       POST                           train
/conversations/<conversation_id:path>/trigger_intent POST                           trigger_intent
/model                                             DELETE                         unload_model
/version                                           GET                            version
2023-08-30 08:19:57,320 [INFO ]  Starting Rasa server on http://0.0.0.0:5005
2023-08-30 08:19:57,320 [DEBUG]  Using the default number of Sanic workers (1).
2023-08-30 08:19:58,135 [DEBUG]  Starting new HTTPS connection (1): api.segment.io:443
2023-08-30 08:19:58,976 [DEBUG]  https://api.segment.io:443 "POST /v1/track HTTP/1.1" 200 21
2023-08-30 08:19:59,027 [DEBUG]  Connected to InMemoryTrackerStore.
2023-08-30 08:19:59,028 [DEBUG]  Connected to lock store 'InMemoryLockStore'.
2023-08-30 08:19:59,028 [DEBUG]  Instantiated NLG to 'TemplatedNaturalLanguageGenerator'.
2023-08-30 08:19:59,029 [INFO ]  Loading model /app/models/20230830-081046-alternating-record.tar.gz...
2023-08-30 08:19:59,712 [DEBUG]  Extracted model to '/tmp/tmp5pj9u1h3'.
2023-08-30 08:19:59,714 [DEBUG]  Node 'nlu_message_converter' loading 'NLUMessageConverter.load' and kwargs: '{}'.
2023-08-30 08:19:59,715 [DEBUG]  Node 'run_WhitespaceTokenizer0' loading 'WhitespaceTokenizer.load' and kwargs: '{}'.
2023-08-30 08:19:59,718 [DEBUG]  Node 'run_RegexFeaturizer1' loading 'RegexFeaturizer.load' and kwargs: '{}'.
2023-08-30 08:19:59,718 [DEBUG]  Resource 'train_RegexFeaturizer1' was requested for reading.
2023-08-30 08:19:59,719 [DEBUG]  Node 'run_RegexEntityExtractor2' loading 'RegexEntityExtractor.load' and kwargs: '{}'.
2023-08-30 08:19:59,720 [DEBUG]  Resource 'train_RegexEntityExtractor2' was requested for reading.
2023-08-30 08:19:59,721 [DEBUG]  Node 'run_LexicalSyntacticFeaturizer3' loading 'LexicalSyntacticFeaturizer.load' and kwargs: '{}'.
2023-08-30 08:19:59,721 [DEBUG]  Resource 'train_LexicalSyntacticFeaturizer3' was requested for reading.
2023-08-30 08:19:59,723 [DEBUG]  Node 'run_CountVectorsFeaturizer4' loading 'CountVectorsFeaturizer.load' and kwargs: '{}'.
2023-08-30 08:19:59,724 [DEBUG]  Resource 'train_CountVectorsFeaturizer4' was requested for reading.
2023-08-30 08:19:59,727 [DEBUG]  Node 'run_CountVectorsFeaturizer5' loading 'CountVectorsFeaturizer.load' and kwargs: '{}'.
2023-08-30 08:19:59,727 [DEBUG]  Resource 'train_CountVectorsFeaturizer5' was requested for reading.
2023-08-30 08:19:59,738 [DEBUG]  Node 'run_DIETClassifier6' loading 'DIETClassifier.load' and kwargs: '{}'.
2023-08-30 08:19:59,738 [DEBUG]  Resource 'train_DIETClassifier6' was requested for reading.
2023-08-30 08:19:59,740 [DEBUG]  Loading the model from /tmp/tmpr4da22h9/train_DIETClassifier6/DIETClassifier.tf_model with finetune_mode=False...
2023-08-30 08:19:59,790 [DEBUG]  Following metrics will be logged during training: 
2023-08-30 08:19:59,790 [DEBUG]    t_loss (total loss)
2023-08-30 08:19:59,791 [DEBUG]    i_acc (intent acc)
2023-08-30 08:19:59,791 [DEBUG]    i_loss (intent loss)
2023-08-30 08:19:59,793 [DEBUG]    e_f1 (entity f1)
2023-08-30 08:19:59,793 [DEBUG]    e_loss (entity loss)
2023-08-30 08:20:29,958 [DEBUG]  Finished loading the model.
2023-08-30 08:20:29,970 [DEBUG]  Node 'run_EntitySynonymMapper7' loading 'EntitySynonymMapper.load' and kwargs: '{}'.
2023-08-30 08:20:29,970 [DEBUG]  Resource 'train_EntitySynonymMapper7' was requested for reading.
2023-08-30 08:20:29,972 [DEBUG]  Node 'run_ResponseSelector8' loading 'ResponseSelector.load' and kwargs: '{}'.
2023-08-30 08:20:29,973 [DEBUG]  Resource 'train_ResponseSelector8' was requested for reading.
2023-08-30 08:20:29,974 [DEBUG]  Failed to load ABCMeta from model storage. Resource 'train_ResponseSelector8' doesn't exist.
2023-08-30 08:20:29,974 [DEBUG]  Resource 'train_ResponseSelector8' was requested for reading.
2023-08-30 08:20:29,975 [DEBUG]  Failed to load ResponseSelector from model storage. Resource 'train_ResponseSelector8' doesn't exist.
2023-08-30 08:20:29,976 [DEBUG]  Node 'run_FallbackClassifier9' loading 'FallbackClassifier.load' and kwargs: '{}'.
2023-08-30 08:20:29,977 [DEBUG]  Node 'run_DucklingEntityExtractor10' loading 'DucklingEntityExtractor.load' and kwargs: '{}'.
2023-08-30 08:20:29,978 [DEBUG]  Node 'run_RegexMessageHandler' loading 'RegexMessageHandler.load' and kwargs: '{}'.
2023-08-30 08:20:29,980 [DEBUG]  Node 'domain_provider' loading 'DomainProvider.load' and kwargs: '{}'.
2023-08-30 08:20:29,986 [DEBUG]  Resource 'domain_provider' was requested for reading.
2023-08-30 08:20:30,004 [DEBUG]  Node 'run_MemoizationPolicy0' loading 'MemoizationPolicy.load' and kwargs: '{}'.
2023-08-30 08:20:30,004 [DEBUG]  Resource 'train_MemoizationPolicy0' was requested for reading.
2023-08-30 08:20:30,005 [DEBUG]  Node 'run_RulePolicy1' loading 'RulePolicy.load' and kwargs: '{}'.
2023-08-30 08:20:30,006 [DEBUG]  Resource 'train_RulePolicy1' was requested for reading.
2023-08-30 08:20:30,007 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' loading 'UnexpecTEDIntentPolicy.load' and kwargs: '{}'.
2023-08-30 08:20:30,008 [DEBUG]  Resource 'train_UnexpecTEDIntentPolicy2' was requested for reading.
2023-08-30 08:20:30,011 [DEBUG]  Loading the model from /tmp/tmpr4da22h9/train_UnexpecTEDIntentPolicy2/unexpected_intent_policy.tf_model with finetune_mode=False...
2023-08-30 08:20:40,773 [DEBUG]  Finished loading the model.
2023-08-30 08:20:40,773 [WARNI]  The UnexpecTED Intent Policy is currently experimental and might change or be removed in the future ðŸ”¬ Please share your feedback on it in the forum (https://forum.rasa.com) to help us make this feature ready for production.
2023-08-30 08:20:40,774 [DEBUG]  Node 'run_TEDPolicy3' loading 'TEDPolicy.load' and kwargs: '{}'.
2023-08-30 08:20:40,774 [DEBUG]  Resource 'train_TEDPolicy3' was requested for reading.
2023-08-30 08:20:40,780 [DEBUG]  Loading the model from /tmp/tmpr4da22h9/train_TEDPolicy3/ted_policy.tf_model with finetune_mode=False...
2023-08-30 08:20:50,361 [DEBUG]  Finished loading the model.
2023-08-30 08:20:50,362 [DEBUG]  Node 'rule_only_data_provider' loading 'RuleOnlyDataProvider.load' and kwargs: '{}'.
2023-08-30 08:20:50,362 [DEBUG]  Resource 'train_RulePolicy1' was requested for reading.
2023-08-30 08:20:50,363 [DEBUG]  Node 'select_prediction' loading 'DefaultPolicyPredictionEnsemble.load' and kwargs: '{}'.
2023-08-30 08:20:50,371 [INFO ]  Rasa server is up and running.
2023-08-30 08:20:50,373 [INFO ]  Enabling coroutine debugging. Loop id 128420224.
2023-08-30 08:20:55,016 [DEBUG]  Issuing ticket for conversation 'user'.
2023-08-30 08:20:55,017 [DEBUG]  Acquiring lock for conversation 'user'.
2023-08-30 08:20:55,017 [DEBUG]  Acquired lock for conversation 'user'.
2023-08-30 08:20:55,018 [DEBUG]  Could not find tracker for conversation ID 'user'.
2023-08-30 08:20:55,019 [DEBUG]  Starting a new session for conversation ID 'user'.
2023-08-30 08:20:55,021 [DEBUG]  Policy prediction ended with events '[]'.
2023-08-30 08:20:55,022 [DEBUG]  Action 'action_session_start' ended with events '[<rasa.shared.core.events.SessionStarted object at 0x7f995ee63d90>, ActionExecuted(action: action_listen, policy: None, confidence: None)]'.
2023-08-30 08:20:55,023 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 08:20:55,024 [DEBUG]  Running graph with inputs: {'__message__': [<rasa.core.channels.channel.UserMessage object at 0x7f999d724910>]}, targets: ['run_RegexMessageHandler'] and ExecutionContext(model_id='3bb02b4afd014417b31fc8b341da2db7', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 08:20:55,026 [DEBUG]  Node 'nlu_message_converter' running 'NLUMessageConverter.convert_user_message'.
2023-08-30 08:20:55,027 [DEBUG]  Node 'run_WhitespaceTokenizer0' running 'WhitespaceTokenizer.process'.
2023-08-30 08:20:55,029 [DEBUG]  Node 'run_RegexFeaturizer1' running 'RegexFeaturizer.process'.
2023-08-30 08:20:55,032 [DEBUG]  Node 'run_RegexEntityExtractor2' running 'RegexEntityExtractor.process'.
2023-08-30 08:20:55,035 [DEBUG]  Node 'run_LexicalSyntacticFeaturizer3' running 'LexicalSyntacticFeaturizer.process'.
2023-08-30 08:20:55,036 [DEBUG]  Node 'run_CountVectorsFeaturizer4' running 'CountVectorsFeaturizer.process'.
2023-08-30 08:20:55,037 [DEBUG]  Node 'run_CountVectorsFeaturizer5' running 'CountVectorsFeaturizer.process'.
2023-08-30 08:20:55,039 [DEBUG]  Node 'run_DIETClassifier6' running 'DIETClassifier.process'.
2023-08-30 08:20:55,356 [DEBUG]  Node 'run_EntitySynonymMapper7' running 'EntitySynonymMapper.process'.
2023-08-30 08:20:55,357 [DEBUG]  Node 'run_ResponseSelector8' running 'ResponseSelector.process'.
2023-08-30 08:20:55,357 [DEBUG]  There is no trained model for 'ResponseSelector': The component is either not trained or didn't receive enough training data.
2023-08-30 08:20:55,358 [DEBUG]  Adding following selector key to message property: default
2023-08-30 08:20:55,358 [DEBUG]  Node 'run_FallbackClassifier9' running 'FallbackClassifier.process'.
2023-08-30 08:20:55,359 [DEBUG]  Node 'run_DucklingEntityExtractor10' running 'DucklingEntityExtractor.process'.
2023-08-30 08:20:55,362 [DEBUG]  Starting new HTTP connection (1): localhost:8000
2023-08-30 08:20:55,363 [ERROR]  Failed to connect to duckling http server. Make sure the duckling server is running/healthy/not stale and the proper host and port are set in the configuration. More information on how to run the server can be found on github: https://github.com/facebook/duckling#quickstart Error: HTTPConnectionPool(host='localhost', port=8000): Max retries exceeded with url: /parse (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f995edd61c0>: Failed to establish a new connection: [Errno 111] Connection refused'))
2023-08-30 08:20:55,364 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 08:20:55,365 [DEBUG]  Node 'run_RegexMessageHandler' running 'RegexMessageHandler.process'.
2023-08-30 08:20:55,365 [DEBUG]  Received user message 'Tell me about Mac Kinh Vu ?' with intent '{'name': 'ask_about_historicalFigure', 'confidence': 1.0}' and entities '[{'entity': 'object', 'start': 14, 'end': 25, 'confidence_entity': 0.999969482421875, 'value': 'culturaltourism:Máº¡c_KÃ­nh_VÅ©', 'extractor': 'DIETClassifier', 'processors': ['EntitySynonymMapper']}]'
2023-08-30 08:20:55,366 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 08:20:55,367 [DEBUG]  Logged UserUtterance - tracker now has 4 events.
2023-08-30 08:20:55,368 [DEBUG]  Validating extracted slots: 
2023-08-30 08:20:55,369 [DEBUG]  Default action 'action_extract_slots' was executed, resulting in 0 events: 
2023-08-30 08:20:55,370 [DEBUG]  Running graph with inputs: {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x7f993f3d68e0>}, targets: ['select_prediction'] and ExecutionContext(model_id='3bb02b4afd014417b31fc8b341da2db7', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 08:20:55,371 [DEBUG]  Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
2023-08-30 08:20:55,372 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 08:20:55,373 [DEBUG]  Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
2023-08-30 08:20:55,374 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
2023-08-30 08:20:55,375 [DEBUG]  There is no memorised next action
2023-08-30 08:20:55,376 [DEBUG]  Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
2023-08-30 08:20:55,377 [DEBUG]  Current tracker state:
[state 1] user text: Tell me about Mac Kinh Vu ? | previous action name: action_listen
2023-08-30 08:20:55,378 [DEBUG]  There is no applicable rule.
2023-08-30 08:20:55,379 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
2023-08-30 08:20:55,380 [DEBUG]  There is a rule for the next action 'action_one_condition'.
2023-08-30 08:20:55,381 [DEBUG]  Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
2023-08-30 08:20:55,389 [DEBUG]  TED predicted 'action_listen' based on user intent.
2023-08-30 08:20:55,390 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
2023-08-30 08:20:55,726 [DEBUG]  Querying for intent `ask_about_historicalFigure`.
2023-08-30 08:20:55,727 [DEBUG]  Query intent index 2 not found in label thresholds - {5: 1.5961803}. Check for `action_unlikely_intent` prediction will be skipped.
2023-08-30 08:20:55,728 [DEBUG]  Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
2023-08-30 08:20:55,729 [DEBUG]  Made prediction using user intent.
2023-08-30 08:20:55,730 [DEBUG]  Added `DefinePrevUserUtteredFeaturization(False)` event.
2023-08-30 08:20:55,731 [DEBUG]  Predicted next action using RulePolicy.
2023-08-30 08:20:55,732 [DEBUG]  Predicted next action 'action_one_condition' with confidence 1.00.
2023-08-30 08:20:55,734 [DEBUG]  Calling action endpoint to run action 'action_one_condition'.
2023-08-30 08:20:55,741 [ERROR]  Failed to run custom action 'action_one_condition'. Couldn't connect to the server at 'http://localhost:5055/webhook'. Is the server running? Error: Cannot connect to host localhost:5055 ssl:default [Connection refused]
2023-08-30 08:20:55,742 [ERROR]  Encountered an exception while running action 'action_one_condition'.Bot will continue, but the actions events are lost. Please check the logs of your action server for more information.
Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 980, in _wrap_create_connection
    return await self._loop.create_connection(*args, **kwargs)  # type: ignore[return-value]  # noqa
  File "uvloop/loop.pyx", line 2039, in create_connection
  File "uvloop/loop.pyx", line 2016, in uvloop.loop.Loop.create_connection
ConnectionRefusedError: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/actions/action.py", line 741, in run
    response: Any = await self.action_endpoint.request(
  File "/opt/venv/lib/python3.8/site-packages/rasa/utils/endpoints.py", line 164, in request
    async with session.request(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/client.py", line 1141, in __aenter__
    self._resp = await self._coro
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/client.py", line 536, in _request
    conn = await self._connector.connect(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 540, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 901, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 1206, in _create_direct_connection
    raise last_exc
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 1175, in _create_direct_connection
    transp, proto = await self._wrap_create_connection(
  File "/opt/venv/lib/python3.8/site-packages/aiohttp/connector.py", line 988, in _wrap_create_connection
    raise client_error(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host localhost:5055 ssl:default [Connection refused]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/processor.py", line 874, in _run_action
    events = await action.run(
  File "/opt/venv/lib/python3.8/site-packages/rasa/core/actions/action.py", line 776, in run
    raise RasaException(
rasa.shared.exceptions.RasaException: Failed to execute custom action 'action_one_condition'. Couldn't connect to the server at 'http://localhost:5055/webhook.
2023-08-30 08:20:55,747 [DEBUG]  Policy prediction ended with events '[<rasa.shared.core.events.DefinePrevUserUtteredFeaturization object at 0x7f995ee33b20>]'.
2023-08-30 08:20:55,747 [DEBUG]  Action 'action_one_condition' ended with events '[]'.
2023-08-30 08:20:55,748 [DEBUG]  Current slot values: 
	session_started_metadata: None
2023-08-30 08:20:55,749 [DEBUG]  Running graph with inputs: {'__tracker__': <rasa.shared.core.trackers.DialogueStateTracker object at 0x7f993f3d68e0>}, targets: ['select_prediction'] and ExecutionContext(model_id='3bb02b4afd014417b31fc8b341da2db7', should_add_diagnostic_data=False, is_finetuning=False, node_name=None).
2023-08-30 08:20:55,750 [DEBUG]  Node 'rule_only_data_provider' running 'RuleOnlyDataProvider.provide'.
2023-08-30 08:20:55,752 [DEBUG]  Node 'domain_provider' running 'DomainProvider.provide_inference'.
2023-08-30 08:20:55,753 [DEBUG]  Node 'run_MemoizationPolicy0' running 'MemoizationPolicy.predict_action_probabilities'.
2023-08-30 08:20:55,754 [DEBUG]  Current tracker state:
[state 1] previous action name: action_listen
2023-08-30 08:20:55,755 [DEBUG]  There is no memorised next action
2023-08-30 08:20:55,756 [DEBUG]  Node 'run_RulePolicy1' running 'RulePolicy.predict_action_probabilities'.
2023-08-30 08:20:55,758 [DEBUG]  Current tracker state:
[state 1] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_listen
[state 2] user intent: ask_about_historicalFigure | user entities: ('object',) | previous action name: action_one_condition
2023-08-30 08:20:55,759 [DEBUG]  There is a rule for the next action 'action_listen'.
2023-08-30 08:20:55,760 [DEBUG]  Node 'run_TEDPolicy3' running 'TEDPolicy.predict_action_probabilities'.
2023-08-30 08:20:55,765 [DEBUG]  TED predicted 'action_listen' based on user intent.
2023-08-30 08:20:55,766 [DEBUG]  Node 'run_UnexpecTEDIntentPolicy2' running 'UnexpecTEDIntentPolicy.predict_action_probabilities'.
2023-08-30 08:20:55,767 [DEBUG]  Skipping predictions for UnexpecTEDIntentPolicy as either there is no event of type `UserUttered`, event's intent is new and not in domain or there is an event of type `ActionExecuted` after the last `UserUttered`.
2023-08-30 08:20:55,767 [DEBUG]  Node 'select_prediction' running 'DefaultPolicyPredictionEnsemble.combine_predictions_from_kwargs'.
2023-08-30 08:20:55,768 [DEBUG]  Predicted next action using RulePolicy.
2023-08-30 08:20:55,769 [DEBUG]  Predicted next action 'action_listen' with confidence 1.00.
2023-08-30 08:20:55,769 [DEBUG]  Policy prediction ended with events '[]'.
2023-08-30 08:20:55,770 [DEBUG]  Action 'action_listen' ended with events '[]'.
2023-08-30 08:20:55,771 [DEBUG]  Deleted lock for conversation 'user'.
